diff --git OLD/audit.spec NEW/audit.spec
index fd7d4c1..4b4e1b5 100644
--- OLD/audit.spec
+++ NEW/audit.spec
@@ -197,7 +197,7 @@ fi
 %{!?_licensedir:%global license %%doc}
 %license lgpl-2.1.txt
 %{_libdir}/libaudit.so.1*
-%{_libdir}/libauparse.*
+%{_libdir}/libauparse.so.0*
 %{_libdir}/libauplugin.so.1*
 %config(noreplace) %attr(640,root,root) /etc/libaudit.conf
 %{_mandir}/man5/libaudit.conf.5.gz
