diff --git OLD/cups.spec NEW/cups.spec
index d450b0d..c96d2b5 100644
--- OLD/cups.spec
+++ NEW/cups.spec
@@ -107,7 +107,6 @@ BuildRequires: make
 BuildRequires: openldap-devel
 BuildRequires: pam-devel
 BuildRequires: pkgconf-pkg-config
-BuildRequires: pkgconfig(avahi-client)
 BuildRequires: pkgconfig(dbus-1)
 BuildRequires: pkgconfig(gnutls)
 BuildRequires: pkgconfig(libsystemd)
@@ -199,7 +198,6 @@ Requires: zlib-devel
 
 %package libs
 Summary: CUPS printing system - libraries
-Requires: %{name}-filesystem = %{epoch}:%{version}-%{release}
 
 %package filesystem
 Summary: CUPS printing system - directory layout
@@ -358,7 +356,7 @@ export CXXFLAGS="$CXXFLAGS $RPM_OPT_FLAGS -DLDAP_DEPRECATED=1"
   --with-access-log-level=actions \
   --with-cupsd-file-perm=0755 \
   --with-dbusdir=%{_sysconfdir}/dbus-1 \
-  --with-dnssd=avahi \
+  --with-dnssd=no \
   --with-log-file-perm=0600 \
   --with-ondemand=systemd \
   --with-pkgconfpath=%{_libdir}/pkgconfig \
@@ -590,7 +588,6 @@ rm -f %{cups_serverbin}/backend/smb
 %{cups_serverbin}/daemon/cups-deviced
 %{cups_serverbin}/daemon/cups-driverd
 %{cups_serverbin}/daemon/cups-exec
-%{cups_serverbin}/backend/dnssd
 %{cups_serverbin}/backend/failover
 %{cups_serverbin}/backend/http
 %{cups_serverbin}/backend/https
@@ -804,11 +801,9 @@ rm -f %{cups_serverbin}/backend/smb
 %attr(0644, root, root)%{_unitdir}/cups-lpd@.service
 
 %files ipptool
-%{_bindir}/ippfind
 %{_bindir}/ipptool
 %dir %{_datadir}/cups/ipptool
 %{_datadir}/cups/ipptool/*
-%{_mandir}/man1/ippfind.1.gz
 %{_mandir}/man1/ipptool.1.gz
 %{_mandir}/man5/ipptoolfile.5.gz
 
