diff --git OLD/cups.spec NEW/cups.spec
index 8129a5c..54cfbce 100644
--- OLD/cups.spec
+++ NEW/cups.spec
@@ -1,5 +1,6 @@
 %global use_alternatives 1
 %global lspp 1
+%global dnssd 0
 
 # {_exec_prefix}/lib/cups is correct, even on x86_64.
 # It is not used for shared objects but for executables.
@@ -105,7 +106,9 @@ BuildRequires: make
 BuildRequires: openldap-devel
 BuildRequires: pam-devel
 BuildRequires: pkgconf-pkg-config
+%if %{dnssd}
 BuildRequires: pkgconfig(avahi-client)
+%endif
 BuildRequires: pkgconfig(dbus-1)
 BuildRequires: pkgconfig(gnutls)
 BuildRequires: pkgconfig(libsystemd)
@@ -197,7 +200,6 @@ Requires: zlib-devel
 
 %package libs
 Summary: CUPS printing system - libraries
-Requires: %{name}-filesystem = %{epoch}:%{version}-%{release}
 
 %package filesystem
 Summary: CUPS printing system - directory layout
@@ -355,7 +357,11 @@ export CXXFLAGS="$CXXFLAGS $RPM_OPT_FLAGS -DLDAP_DEPRECATED=1"
   --with-access-log-level=actions \
   --with-cupsd-file-perm=0755 \
   --with-dbusdir=%{_sysconfdir}/dbus-1 \
+%if %{dnssd}
   --with-dnssd=avahi \
+%else
+  --with-dnssd=no \
+%endif
   --with-log-file-perm=0600 \
   --with-ondemand=systemd \
   --with-pkgconfpath=%{_libdir}/pkgconfig \
@@ -597,7 +603,9 @@ rm -f %{cups_serverbin}/backend/smb
 %{cups_serverbin}/daemon/cups-deviced
 %{cups_serverbin}/daemon/cups-driverd
 %{cups_serverbin}/daemon/cups-exec
+%if %{dnssd}
 %{cups_serverbin}/backend/dnssd
+%endif
 %{cups_serverbin}/backend/failover
 %{cups_serverbin}/backend/http
 %{cups_serverbin}/backend/https
@@ -811,11 +819,15 @@ rm -f %{cups_serverbin}/backend/smb
 %attr(0644, root, root)%{_unitdir}/cups-lpd@.service
 
 %files ipptool
+%if %{dnssd}
 %{_bindir}/ippfind
+%endif
 %{_bindir}/ipptool
 %dir %{_datadir}/cups/ipptool
 %{_datadir}/cups/ipptool/*
+%if %{dnssd}
 %{_mandir}/man1/ippfind.1.gz
+%endif
 %{_mandir}/man1/ipptool.1.gz
 %{_mandir}/man5/ipptoolfile.5.gz
 
