--- dbus.spec	2024-10-08 00:44:00.141571742 +0800
+++ dbus.spec.new	2024-10-08 01:03:27.883146114 +0800
@@ -100,6 +100,7 @@
 Summary:        D-BUS message bus
 Conflicts:      fedora-release < %{fedora_release_version}
 Conflicts:      generic-release < %{generic_release_version}
+Conflicts:      dbus-gdm
 Requires:       libselinux%{?_isa} >= %{libselinux_version}
 Requires:       dbus-common = %{epoch}:%{version}-%{release}
 Requires:       dbus-libs%{?_isa} = %{epoch}:%{version}-%{release}
@@ -111,6 +112,23 @@
 used both for the system-wide message bus service, and as a
 per-user-login-session messaging facility.
 
+%package gdm
+Summary:        D-BUS message bus for gdm
+Conflicts:      fedora-release < %{fedora_release_version}
+Conflicts:      generic-release < %{generic_release_version}
+Conflicts:      dbus-daemon
+Requires:       libselinux%{?_isa} >= %{libselinux_version}
+Requires:       dbus-common = %{epoch}:%{version}-%{release}
+Requires:       dbus-libs%{?_isa} = %{epoch}:%{version}-%{release}
+%{?sysusers_requires_compat}
+
+%description gdm
+D-BUS is a system for sending messages between applications. It is
+used both for the system-wide message bus service, and as a
+per-user-login-session messaging facility.
+
+This package is for use by GDM.
+
 %package tools
 Summary:        D-BUS Tools and Utilities
 Requires:       dbus-libs%{?_isa} = %{epoch}:%{version}-%{release}
@@ -397,6 +415,23 @@
 %{_unitdir}/dbus-daemon.service
 %{_userunitdir}/dbus-daemon.service
 
+%files gdm
+# Strictly speaking, we could remove the COPYING from this subpackage and 
+# just have it be in libs, because dbus Requires dbus-libs.
+%{!?_licensedir:%global license %%doc}
+%license COPYING
+%doc AUTHORS CONTRIBUTING.md NEWS README
+%exclude %{_pkgdocdir}/api
+%exclude %{_pkgdocdir}/diagram.*
+%exclude %{_pkgdocdir}/introspect.*
+%exclude %{_pkgdocdir}/system-activation.txt
+%exclude %{_pkgdocdir}/*.html
+%ghost %dir /run/%{name}
+%{_bindir}/dbus-daemon
+%{_bindir}/dbus-run-session
+%{_mandir}/man1/dbus-daemon.1*
+%{_mandir}/man1/dbus-run-session.1*
+
 %files tools
 %{!?_licensedir:%global license %%doc}
 %license COPYING
