diff --git OLD/desktop-file-utils.spec NEW/desktop-file-utils.spec
index 7161858..22af31b 100644
--- OLD/desktop-file-utils.spec
+++ NEW/desktop-file-utils.spec
@@ -13,7 +13,8 @@ BuildRequires: gcc
 BuildRequires: glib2-devel
 BuildRequires: emacs
 BuildRequires: meson
-Requires: emacs-filesystem
+
+Requires: (desktop-file-emacs = %{version}-%{release} if emacs-common)
 
 %description
 .desktop files are used to describe an application for inclusion in
@@ -24,6 +25,17 @@ which installs a desktop file to the standard directory, optionally
 fixing it up in the process.
 
 
+%package -n desktop-file-emacs
+Summary: Emacs support for desktop-file-utils
+Requires: %{name} = %{version}-%{release}
+Requires: emacs-filesystem
+
+BuildArch: noarch
+
+%description -n desktop-file-emacs
+This package contains the Emacs support for desktop-file-utils.
+
+
 %prep
 %autosetup -p1
 
@@ -53,6 +65,8 @@ update-desktop-database &> /dev/null || :
 %{_mandir}/man1/desktop-file-validate.1*
 %{_mandir}/man1/update-desktop-database.1*
 %{_mandir}/man1/desktop-file-edit.1*
+
+%files -n desktop-file-emacs
 %{_emacs_sitestartdir}/desktop-entry-mode-init.el
 %ghost %{_emacs_sitestartdir}/desktop-entry-mode-init.elc
 %{_emacs_sitelispdir}/desktop-file-utils/
