diff --git OLD/dnf5.spec NEW/dnf5.spec
index 5742da7..4381dc3 100644
--- OLD/dnf5.spec
+++ NEW/dnf5.spec
@@ -13,9 +13,11 @@
 %bcond dnf5_obsoletes_dnf %[0%{?fedora} > 40 || 0%{?rhel} > 10]
 %bcond_without acl
 %bcond_without comps
-%if 0%{?rhel} >= 11
+%if 0%{?fedora} > 40 || 0%{?rhel} > 10
+%bcond_with    microdnf
 %bcond_with    modulemd
 %else
+%bcond_without microdnf
 %bcond_without modulemd
 %endif
 %bcond_without systemd
@@ -98,8 +100,9 @@ Requires:       coreutils
 %if 0%{?fedora} > 41
 Recommends:     (libdnf5-plugin-expired-pgp-keys if gnupg2)
 %endif
+Requires:       (%{name}-zsh-completion = %{version}-%{release} if zsh)
 
-%if 0%{?fedora} || 0%{?rhel} > 10
+%if %{with microdnf}
 Provides:       microdnf = %{version}-%{release}
 Obsoletes:      microdnf < 4
 %endif
@@ -108,9 +111,6 @@ Obsoletes:      microdnf < 4
 Provides:       dnf = %{version}-%{release}
 Obsoletes:      dnf < 5
 
-Provides:       yum = %{version}-%{release}
-Obsoletes:      yum < 5
-
 Conflicts:      python3-dnf-plugins-core < 4.7.0
 %endif
 
@@ -316,12 +316,11 @@ It supports RPM packages%{?with_modulemd:, modulemd modules,} and comps groups &
 %{_bindir}/dnf5
 %if %{with dnf5_obsoletes_dnf}
 %{_bindir}/dnf
-%{_bindir}/yum
 %endif
 %{_unitdir}/dnf*-makecache.service
 %{_unitdir}/dnf*-makecache.timer
 
-%if 0%{?fedora} || 0%{?rhel} > 10
+%if %{with microdnf}
 %{_bindir}/microdnf
 %endif
 
@@ -339,9 +338,6 @@ It supports RPM packages%{?with_modulemd:, modulemd modules,} and comps groups &
 %dir %{_datadir}/bash-completion/
 %dir %{_datadir}/bash-completion/completions/
 %{_datadir}/bash-completion/completions/dnf*
-%dir %{_datadir}/zsh/
-%dir %{_datadir}/zsh/site-functions/
-%{_datadir}/zsh/site-functions/_dnf5
 %license COPYING.md
 %license gpl-2.0.txt
 %doc AUTHORS.md CHANGELOG.md CONTRIBUTING.md README.md
@@ -465,6 +461,20 @@ It supports RPM packages%{?with_modulemd:, modulemd modules,} and comps groups &
 %exclude %{_datadir}/dnf5/aliases.d/compatibility-reposync.conf
 %endif
 
+# ========== dnf5-zsh-completion ==========
+%package zsh-completion
+Summary:        Zsh completion scripts for %{name}
+Requires:       %{name} = %{version}-%{release}
+Requires:       zsh
+
+BuildArch:      noarch
+
+%description zsh-completion
+Zsh completion scripts for %{name}
+
+%files zsh-completion
+%{_datadir}/zsh/site-functions/*
+
 # ========== libdnf5 ==========
 %package -n libdnf5
 Summary:        Package management library
@@ -1124,7 +1134,6 @@ DNF5 plugin for working with RPM package manifest files.
 
 %if %{with dnf5_obsoletes_dnf}
 ln -sr %{buildroot}%{_bindir}/dnf5 %{buildroot}%{_bindir}/dnf
-ln -sr %{buildroot}%{_bindir}/dnf5 %{buildroot}%{_bindir}/yum
 ln -sr %{buildroot}%{_datadir}/bash-completion/completions/dnf5 %{buildroot}%{_datadir}/bash-completion/completions/dnf
 %if %{with man}
     for file in %{buildroot}%{_mandir}/man[578]/dnf5[-.]*; do
@@ -1160,7 +1169,7 @@ touch %{buildroot}%{_sysconfdir}/dnf/versionlock.toml
 mkdir -p %{buildroot}%{_sharedstatedir}/dnf
 touch %{buildroot}%{_sharedstatedir}/dnf/system-repo.lock
 
-%if 0%{?fedora} || 0%{?rhel} > 10
+%if %{with microdnf}
 ln -sr %{buildroot}%{_bindir}/dnf5 %{buildroot}%{_bindir}/microdnf
 %endif
 
