diff --git OLD/dnf5.spec NEW/dnf5.spec
index be49f55..6d5d0df 100644
--- OLD/dnf5.spec
+++ NEW/dnf5.spec
@@ -13,9 +13,11 @@
 %bcond dnf5_obsoletes_dnf %[0%{?fedora} > 40 || 0%{?rhel} > 10]
 %bcond_without acl
 %bcond_without comps
-%if 0%{?rhel} >= 11
+%if 0%{?fedora} > 40 || 0%{?rhel} > 10
+%bcond_with    microdnf
 %bcond_with    modulemd
 %else
+%bcond_without microdnf
 %bcond_without modulemd
 %endif
 %bcond_without systemd
@@ -94,8 +96,9 @@ Requires:       coreutils
 %if 0%{?fedora} > 41
 Recommends:     (libdnf5-plugin-expired-pgp-keys if gnupg2)
 %endif
+Requires:       (%{name}-zsh-completion = %{version}-%{release} if zsh)
 
-%if 0%{?fedora} || 0%{?rhel} > 10
+%if %{with microdnf}
 Provides:       microdnf = %{version}-%{release}
 Obsoletes:      microdnf < 4
 %endif
@@ -104,9 +107,6 @@ Obsoletes:      microdnf < 4
 Provides:       dnf = %{version}-%{release}
 Obsoletes:      dnf < 5
 
-Provides:       yum = %{version}-%{release}
-Obsoletes:      yum < 5
-
 Conflicts:      python3-dnf-plugins-core < 4.7.0
 %endif
 
@@ -312,12 +312,11 @@ It supports RPM packages%{?with_modulemd:, modulemd modules,} and comps groups &
 %{_bindir}/dnf5
 %if %{with dnf5_obsoletes_dnf}
 %{_bindir}/dnf
-%{_bindir}/yum
 %endif
 %{_unitdir}/dnf*-makecache.service
 %{_unitdir}/dnf*-makecache.timer
 
-%if 0%{?fedora} || 0%{?rhel} > 10
+%if %{with microdnf}
 %{_bindir}/microdnf
 %endif
 
@@ -332,8 +331,7 @@ It supports RPM packages%{?with_modulemd:, modulemd modules,} and comps groups &
 %dir %{_sysconfdir}/dnf/dnf5-plugins
 %doc %{_libdir}/dnf5/plugins/README
 %dir %{_libdir}/libdnf5/plugins
-%{bash_completions_dir}/dnf*
-%{zsh_completions_dir}/_dnf5
+%{bash_completions_dir}/*
 %license COPYING.md
 %license gpl-2.0.txt
 %doc AUTHORS.md CHANGELOG.md CONTRIBUTING.md README.md
@@ -459,6 +457,20 @@ It supports RPM packages%{?with_modulemd:, modulemd modules,} and comps groups &
 %exclude %{_datadir}/dnf5/aliases.d/compatibility-reposync.conf
 %endif
 
+# ========== dnf5-zsh-completion ==========
+%package zsh-completion
+Summary:        Zsh completion scripts for %{name}
+Requires:       %{name} = %{version}-%{release}
+Requires:       zsh
+
+BuildArch:      noarch
+
+%description zsh-completion
+Zsh completion scripts for %{name}
+
+%files zsh-completion
+%{zsh_completions_dir}/*
+
 # ========== libdnf5 ==========
 %package -n libdnf5
 Summary:        Package management library
@@ -1119,7 +1131,6 @@ DNF5 plugin for working with RPM package manifest files.
 
 %if %{with dnf5_obsoletes_dnf}
 ln -sr %{buildroot}%{_bindir}/dnf5 %{buildroot}%{_bindir}/dnf
-ln -sr %{buildroot}%{_bindir}/dnf5 %{buildroot}%{_bindir}/yum
 ln -sr %{buildroot}%{bash_completions_dir}/dnf5 %{buildroot}%{bash_completions_dir}/dnf
 %if %{with man}
     for file in %{buildroot}%{_mandir}/man[578]/dnf5[-.]*; do
@@ -1155,7 +1166,7 @@ touch %{buildroot}%{_sysconfdir}/dnf/versionlock.toml
 mkdir -p %{buildroot}%{_sharedstatedir}/dnf
 touch %{buildroot}%{_sharedstatedir}/dnf/system-repo.lock
 
-%if 0%{?fedora} || 0%{?rhel} > 10
+%if %{with microdnf}
 ln -sr %{buildroot}%{_bindir}/dnf5 %{buildroot}%{_bindir}/microdnf
 %endif
 
