diff --git OLD/dracut.spec NEW/dracut.spec
index f0f860c..c9aea38 100644
--- OLD/dracut.spec
+++ NEW/dracut.spec
@@ -190,6 +190,13 @@ This package provides a dracut module to build an initramfs, but store most file
 in a squashfs image, result in a smaller initramfs size and reduce runtime memory
 usage.
 
+%package devel
+Summary: Development files for dracut
+Requires: %{name} = %{version}-%{release}
+
+%description devel
+This package provides the development files for dracut.
+
 %prep
 %autosetup -n %{name}-ng-%{version} -S git_am
 cp %{SOURCE1} .
@@ -282,8 +289,6 @@ echo 'dracut_rescue_image="yes"' > $RPM_BUILD_ROOT%{dracutlibdir}/dracut.conf.d/
 %{dracutlibdir}/dracut.conf.d/01-dist.conf
 %dir %{_sysconfdir}/dracut.conf.d
 %dir %{dracutlibdir}/dracut.conf.d
-%dir %{_datadir}/pkgconfig
-%{_datadir}/pkgconfig/dracut.pc
 
 %if %{with doc}
 %{_mandir}/man8/dracut.8*
@@ -482,6 +487,9 @@ echo 'dracut_rescue_image="yes"' > $RPM_BUILD_ROOT%{dracutlibdir}/dracut.conf.d/
 %{dracutlibdir}/dracut.conf.d/02-rescue.conf
 %{_prefix}/lib/kernel/install.d/51-dracut-rescue.install
 
+%files devel
+%{_datadir}/pkgconfig/dracut.pc
+
 %changelog
 * Thu Jul 24 2025 Pavel Valena <pvalena@redhat.com> - 107-2
 - Revert "feat(hwdb): add hwdb module to install hwdb.bin on demand"
