diff --git OLD/dracut.spec NEW/dracut.spec
index 9a089dc..1bec154 100644
--- OLD/dracut.spec
+++ NEW/dracut.spec
@@ -195,6 +195,13 @@ This package provides a dracut module to build an initramfs, but store most file
 in a squashfs image, result in a smaller initramfs size and reduce runtime memory
 usage.
 
+%package devel
+Summary: Development files for dracut
+Requires: %{name} = %{version}-%{release}
+
+%description devel
+This package provides the development files for dracut.
+
 %prep
 %autosetup -n %{name}-ng-%{version} -S git_am
 cp %{SOURCE1} .
@@ -287,8 +294,6 @@ echo 'dracut_rescue_image="yes"' > $RPM_BUILD_ROOT%{dracutlibdir}/dracut.conf.d/
 %{dracutlibdir}/dracut.conf.d/01-dist.conf
 %dir %{_sysconfdir}/dracut.conf.d
 %dir %{dracutlibdir}/dracut.conf.d
-%dir %{_datadir}/pkgconfig
-%{_datadir}/pkgconfig/dracut.pc
 
 %if %{with doc}
 %{_mandir}/man8/dracut.8*
@@ -489,6 +494,9 @@ echo 'dracut_rescue_image="yes"' > $RPM_BUILD_ROOT%{dracutlibdir}/dracut.conf.d/
 %{dracutlibdir}/dracut.conf.d/02-rescue.conf
 %{_prefix}/lib/kernel/install.d/51-dracut-rescue.install
 
+%files devel
+%{_datadir}/pkgconfig/dracut.pc
+
 %changelog
 * Thu Mar 12 2026 Pavel Valena <pvalena@redhat.com> - 108-6
 - fix(dracut): use grep -q/-s to silence output/error
