diff --git OLD/evolution-data-server.spec NEW/evolution-data-server.spec
index c823a88..0e9d7e8 100644
--- OLD/evolution-data-server.spec
+++ NEW/evolution-data-server.spec
@@ -1,17 +1,15 @@
 %undefine __cmake_in_source_build
 
-%define ldap_support 1
+%define ldap_support 0
 %define static_ldap 0
 %define krb5_support 1
 %define largefile_support 1
 
-# disable for i686, because libphonenumber 8.12.57 is not built for it
-%ifarch i686
+# disable libphonenumber
 %global phonenum_support 0
-%else
-# enabled only for Fedora
-%global phonenum_support 0%{?fedora}
-%endif
+
+# disable libgweather
+%global weather_support 0
 
 # Coverity scan can override this to 0, to skip checking in gtk-doc generated code
 %{!?with_docs: %global with_docs 1}
@@ -33,7 +31,6 @@
 %define libsoup_version 3.1.1
 %define nss_version 3.14
 %define sqlite_version 3.7.17
-%define webkit2gtk_version 2.34.0
 %define webkit2gtk4_version 2.36.0
 %define json_glib_version 1.0.4
 %define uuid_version 2.0
@@ -102,7 +99,9 @@ BuildRequires: pkgconfig(icu-i18n)
 BuildRequires: pkgconfig(gtk+-3.0) >= %{gtk3_version}
 BuildRequires: pkgconfig(gtk4) >= %{gtk4_version}
 BuildRequires: pkgconfig(goa-1.0) >= %{goa_version}
+%if %{weather_support}
 BuildRequires: pkgconfig(gweather4) >= %{libgweather_version}
+%endif
 BuildRequires: pkgconfig(libical-glib) >= %{libical_version}
 BuildRequires: pkgconfig(libsecret-unstable) >= %{libsecret_version}
 BuildRequires: pkgconfig(libsoup-3.0) >= %{libsoup_version}
@@ -112,7 +111,6 @@ BuildRequires: pkgconfig(nss) >= %{nss_version}
 BuildRequires: pkgconfig(sqlite3) >= %{sqlite_version}
 BuildRequires: pkgconfig(uuid) >= %{uuid_version}
 %if %{with_webkitgtk}
-BuildRequires: pkgconfig(webkit2gtk-4.1) >= %{webkit2gtk_version}
 BuildRequires: pkgconfig(webkitgtk-6.0) >= %{webkit2gtk4_version}
 %endif
 BuildRequires: pkgconfig(json-glib-1.0) >= %{json_glib_version}
@@ -151,13 +149,14 @@ Summary: Development files for building against %{name}
 Requires: %{name}%{?_isa} = %{version}-%{release}
 
 Requires: pkgconfig(goa-1.0) >= %{goa_version}
+%if %{weather_support}
 Requires: pkgconfig(gweather4) >= %{libgweather_version}
+%endif
 Requires: pkgconfig(libical-glib) >= %{libical_version}
 Requires: pkgconfig(libsecret-unstable) >= %{libsecret_version}
 Requires: pkgconfig(libsoup-3.0) >= %{libsoup_version}
 Requires: pkgconfig(sqlite3) >= %{sqlite_version}
 %if %{with_webkitgtk}
-Requires: pkgconfig(webkit2gtk-4.1) >= %{webkit2gtk_version}
 Requires: pkgconfig(webkitgtk-6.0) >= %{webkit2gtk4_version}
 %endif
 Requires: pkgconfig(json-glib-1.0) >= %{json_glib_version}
@@ -258,6 +257,12 @@ fi
 %define phonenum_flags -DWITH_PHONENUMBER=OFF
 %endif
 
+%if %{weather_support}
+%define weather_flags -DENABLE_WEATHER=ON
+%else
+%define weather_flags -DENABLE_WEATHER=OFF
+%endif
+
 %define ssl_flags -DENABLE_SMIME=ON
 
 %if %{with_docs}
@@ -267,7 +272,7 @@ fi
 %endif
 
 %if %{with_webkitgtk}
-%define webkitgtk_flags -DENABLE_OAUTH2_WEBKITGTK=ON -DENABLE_OAUTH2_WEBKITGTK4=ON
+%define webkitgtk_flags -DENABLE_OAUTH2_WEBKITGTK=OFF -DENABLE_OAUTH2_WEBKITGTK4=ON
 %else
 %define webkitgtk_flags -DENABLE_OAUTH2_WEBKITGTK=OFF -DENABLE_OAUTH2_WEBKITGTK4=OFF
 %endif
@@ -289,7 +294,7 @@ export CFLAGS="$RPM_OPT_FLAGS -DLDAP_DEPRECATED -fPIC -I%{_includedir}/et -Wno-d
 	-DWITH_LIBDB=OFF \
         -DWITH_SYSTEMDUSERUNITDIR=%{_userunitdir} \
 	%ldap_flags %krb5_flags %ssl_flags %webkitgtk_flags \
-	%largefile_flags %gtkdoc_flags %phonenum_flags \
+	%largefile_flags %gtkdoc_flags %phonenum_flags %weather_flags \
 	-DINCLUDE_INSTALL_DIR:PATH=%{_includedir} \
 	-DLIB_INSTALL_DIR:PATH=%{_libdir} \
 	-DSYSCONF_INSTALL_DIR:PATH=%{_sysconfdir} \
@@ -427,13 +432,17 @@ find $RPM_BUILD_ROOT -name '*.so.*' -exec chmod +x {} \;
 %{credential_modules_dir}/module-credentials-goa.so
 %{ebook_backends_dir}/libebookbackendcarddav.so
 %{ebook_backends_dir}/libebookbackendfile.so
+%if %{ldap_support}
 %{ebook_backends_dir}/libebookbackendldap.so
+%endif
 %{ecal_backends_dir}/libecalbackendcaldav.so
 %{ecal_backends_dir}/libecalbackendcontacts.so
 %{ecal_backends_dir}/libecalbackendfile.so
 %{ecal_backends_dir}/libecalbackendgtasks.so
 %{ecal_backends_dir}/libecalbackendhttp.so
+%if %{weather_support}
 %{ecal_backends_dir}/libecalbackendweather.so
+%endif
 %{ecal_backends_dir}/libecalbackendwebdavnotes.so
 %{modules_dir}/module-cache-reaper.so
 %{modules_dir}/module-google-backend.so
