diff --git OLD/exiv2.spec NEW/exiv2.spec
index 86a15a6..b5b3884 100644
--- OLD/exiv2.spec
+++ NEW/exiv2.spec
@@ -1,3 +1,5 @@
+%bcond_with inih
+
 Name:           exiv2
 Version:        0.28.6
 Release:        %autorelease
@@ -30,7 +32,9 @@ BuildRequires:  gcc-c++
 BuildRequires:  gettext
 BuildRequires:  pkgconfig
 BuildRequires:  pkgconfig(expat)
+%if %{with inih}
 BuildRequires:  pkgconfig(INIReader)
+%endif
 BuildRequires:  pkgconfig(libbrotlidec)
 BuildRequires:  pkgconfig(zlib)
 # docs
@@ -106,7 +110,13 @@ API documentation for %{name}.
   -DCMAKE_INSTALL_DOCDIR="%{_pkgdocdir}" \
   -DEXIV2_BUILD_DOC:BOOL=ON \
   -DEXIV2_BUILD_SAMPLES:BOOL=OFF \
-  -DEXIV2_ENABLE_NLS:BOOL=ON
+  -DEXIV2_ENABLE_NLS:BOOL=ON \
+%if %{with inih}
+  -DEXIV2_ENABLE_INIH:BOOL=ON \
+%else
+  -DEXIV2_ENABLE_INIH:BOOL=OFF \
+%endif
+  %{nil}
 %cmake_build
 %cmake_build --target doc
 
