diff --git OLD/flatpak.spec NEW/flatpak.spec
index 1f6ce3a..a61c35f 100644
--- OLD/flatpak.spec
+++ NEW/flatpak.spec
@@ -8,8 +8,8 @@
 %global wayland_scanner_version 1.15
 %global xdg_portal_version 1.7.0
 
-# Disable parental control for RHEL builds
-%bcond malcontent %[!0%{?rhel}]
+# Disable malcontent for all builds
+%bcond malcontent 0
 
 Name:           flatpak
 Version:        1.17.7
@@ -81,8 +81,10 @@ Requires:       ostree-libs%{?_isa} >= %{ostree_version}
 Requires:       /usr/bin/fusermount3
 Requires:       /usr/bin/xdg-dbus-proxy
 # https://fedoraproject.org/wiki/SELinux/IndependentPolicy
-Requires:       (flatpak-selinux = %{?epoch:%{epoch}:}%{version}-%{release} if selinux-policy-targeted)
+Requires:       (%{name}-selinux = %{?epoch:%{epoch}:}%{version}-%{release} if selinux-policy-targeted)
 Requires:       %{name}-session-helper%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:       (%{name}-fish-completion = %{?epoch:%{epoch}:}%{version}-%{release} if fish)
+Requires:       (%{name}-zsh-completion = %{?epoch:%{epoch}:}%{version}-%{release} if zsh)
 Recommends:     p11-kit-server
 
 # Make sure the document portal is installed
@@ -135,6 +137,26 @@ Requires:       systemd
 This package contains the org.freedesktop.Flatpak user D-Bus service
 that's used by %{name} and other packages.
 
+%package fish-completion
+Summary:        Fish shell completion for %{name}
+Requires:       %{name} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:       fish
+
+BuildArch:      noarch
+
+%description fish-completion
+This package contains fish shell completion for %{name}.
+
+%package zsh-completion
+Summary:        Zsh shell completion for %{name}
+Requires:       %{name} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:       zsh
+
+BuildArch:      noarch
+
+%description zsh-completion
+This package contains zsh shell completion for %{name}.
+
 %package tests
 Summary:        Tests for %{name}
 Requires:       %{name}%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
@@ -228,11 +250,9 @@ fi
 %{_datadir}/dbus-1/services/org.freedesktop.portal.Flatpak.service
 %{_datadir}/dbus-1/system.d/org.freedesktop.Flatpak.SystemHelper.conf
 %{_datadir}/dbus-1/system-services/org.freedesktop.Flatpak.SystemHelper.service
-%{_datadir}/fish/
 %{_datadir}/%{name}
 %{_datadir}/polkit-1/actions/org.freedesktop.Flatpak.policy
 %{_datadir}/polkit-1/rules.d/org.freedesktop.Flatpak.rules
-%{_datadir}/zsh/site-functions
 %{_libexecdir}/flatpak-oci-authenticator
 %{_libexecdir}/flatpak-portal
 %{_libexecdir}/flatpak-system-helper
@@ -268,6 +288,7 @@ fi
 %files devel
 %{_datadir}/gir-1.0/Flatpak-1.0.gir
 %{_datadir}/gtk-doc/
+%{_datadir}/selinux/devel/include/distributed/flatpak.if
 %{_includedir}/%{name}/
 %{_libdir}/libflatpak.so
 %{_libdir}/pkgconfig/%{name}.pc
@@ -279,7 +300,6 @@ fi
 
 %files selinux
 %{_datadir}/selinux/packages/flatpak.pp.bz2
-%{_datadir}/selinux/devel/include/distributed/flatpak.if
 
 %files session-helper
 %license COPYING
@@ -288,6 +308,12 @@ fi
 %{_libexecdir}/flatpak-session-helper
 %{_userunitdir}/flatpak-session-helper.service
 
+%files fish-completion
+%{_datadir}/fish/*/*
+
+%files zsh-completion
+%{_datadir}/zsh/site-functions/*
+
 %files tests
 %{_datadir}/installed-tests
 %{_libexecdir}/installed-tests
