--- flatpak.spec	2025-02-08 20:48:14.025505231 +0800
+++ flatpak.spec.new	2025-04-14 00:29:11.084111835 +0800
@@ -7,8 +7,8 @@
 %global wayland_protocols_version 1.32
 %global wayland_scanner_version 1.15
 
-# Disable parental control for RHEL builds
-%bcond malcontent %[!0%{?rhel}]
+# Disable malcontent for all builds
+%bcond malcontent 0
 
 Name:           flatpak
 Version:        1.15.12
@@ -86,8 +86,9 @@
 Requires:       /usr/bin/fusermount3
 Requires:       /usr/bin/xdg-dbus-proxy
 # https://fedoraproject.org/wiki/SELinux/IndependentPolicy
-Requires:       (flatpak-selinux = %{?epoch:%{epoch}:}%{version}-%{release} if selinux-policy-targeted)
+Requires:       (%{name}-selinux = %{?epoch:%{epoch}:}%{version}-%{release} if selinux-policy-targeted)
 Requires:       %{name}-session-helper%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:       (%{name}-fish-completion = %{?epoch:%{epoch}:}%{version}-%{release} if fish)
 Recommends:     p11-kit-server
 
 # Make sure the document portal is installed
@@ -140,6 +141,16 @@
 This package contains the org.freedesktop.Flatpak user D-Bus service
 that's used by %{name} and other packages.
 
+%package fish-completion
+Summary:        Fish shell completion for %{name}
+Requires:       %{name}%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:       fish
+
+BuildArch:      noarch
+
+%description fish-completion
+This package contains fish shell completion for %{name}.
+
 %package tests
 Summary:        Tests for %{name}
 Requires:       %{name}%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
@@ -161,6 +172,7 @@
     -Dinstalled_tests=true \
     -Dsystem_bubblewrap=/usr/bin/bwrap \
     -Dsystem_dbus_proxy=/usr/bin/xdg-dbus-proxy \
+    -Ddbus_config_dir=%{_datadir}/dbus-1/system.d \
     -Dtmpfilesdir=%{_tmpfilesdir} \
 %if %{with malcontent}
     -Dmalcontent=enabled \
@@ -230,8 +242,8 @@
 %{_datadir}/dbus-1/interfaces/org.freedesktop.Flatpak.Authenticator.xml
 %{_datadir}/dbus-1/services/org.flatpak.Authenticator.Oci.service
 %{_datadir}/dbus-1/services/org.freedesktop.portal.Flatpak.service
+%{_datadir}/dbus-1/system.d/org.freedesktop.Flatpak.SystemHelper.conf
 %{_datadir}/dbus-1/system-services/org.freedesktop.Flatpak.SystemHelper.service
-%{_datadir}/fish/
 %{_datadir}/%{name}
 %{_datadir}/polkit-1/actions/org.freedesktop.Flatpak.policy
 %{_datadir}/polkit-1/rules.d/org.freedesktop.Flatpak.rules
@@ -250,7 +262,6 @@
 %{_mandir}/man5/flatpak-remote.5*
 %{_mandir}/man5/flatpakref.5*
 %{_mandir}/man5/flatpakrepo.5*
-%{_sysconfdir}/dbus-1/system.d/org.freedesktop.Flatpak.SystemHelper.conf
 %dir %{_sysconfdir}/flatpak
 %{_sysconfdir}/flatpak/remotes.d
 %{_sysconfdir}/profile.d/flatpak.csh
@@ -270,6 +281,7 @@
 %files devel
 %{_datadir}/gir-1.0/Flatpak-1.0.gir
 %{_datadir}/gtk-doc/
+%{_datadir}/selinux/devel/include/contrib/flatpak.if
 %{_includedir}/%{name}/
 %{_libdir}/libflatpak.so
 %{_libdir}/pkgconfig/%{name}.pc
@@ -281,7 +293,6 @@
 
 %files selinux
 %{_datadir}/selinux/packages/flatpak.pp.bz2
-%{_datadir}/selinux/devel/include/contrib/flatpak.if
 
 %files session-helper
 %license COPYING
@@ -290,6 +301,9 @@
 %{_libexecdir}/flatpak-session-helper
 %{_userunitdir}/flatpak-session-helper.service
 
+%files fish-completion
+%{_datadir}/fish/
+
 %files tests
 %{_datadir}/installed-tests
 %{_libexecdir}/installed-tests
