diff --git OLD/fwupd.spec NEW/fwupd.spec
index b8f4307..3b03f6e 100644
--- OLD/fwupd.spec
+++ NEW/fwupd.spec
@@ -115,6 +115,8 @@ Requires: libxmlb%{?_isa} >= %{libxmlb_version}
 Requires: libusb1%{?_isa} >= %{libusb_version}
 Requires: shared-mime-info
 
+Requires: (%{name}-fish-completion = %{version}-%{release} if fish)
+
 Obsoletes: dbxtool < 9
 Provides: dbxtool
 
@@ -140,6 +142,15 @@ Recommends: %{name}-plugin-uefi-capsule-data
 %description
 fwupd is a daemon to allow session software to update device firmware.
 
+%package fish-completion
+Summary: Fish completion for fwupdmgr
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+BuildArch: noarch
+
+%description fish-completion
+Fish completion for command fwupdmgr.
+
 %package devel
 Summary: Development package for %{name}
 Requires: %{name}%{?_isa} = %{version}-%{release}
@@ -283,7 +294,6 @@ systemctl --no-reload preset fwupd-refresh.timer &>/dev/null || :
 %{_datadir}/dbus-1/system.d/org.freedesktop.fwupd.conf
 %{_datadir}/bash-completion/completions/fwupdmgr
 %{_datadir}/bash-completion/completions/fwupdtool
-%{_datadir}/fish/vendor_completions.d/fwupdmgr.fish
 %dir %{_datadir}/fwupd
 %dir %{_datadir}/fwupd/metainfo
 %{_datadir}/fwupd/metainfo/org.freedesktop.fwupd*.metainfo.xml
@@ -324,6 +334,9 @@ systemctl --no-reload preset fwupd-refresh.timer &>/dev/null || :
 %{_libdir}/fwupd-%{version}/libfwupd*.so
 %ghost %{_localstatedir}/lib/fwupd/gnupg
 
+%files fish-completion
+%{_datadir}/fish/vendor_completions.d/fwupdmgr.fish
+
 %if 0%{?have_modem_manager}
 %files plugin-modem-manager
 %{_libdir}/fwupd-%{version}/libfu_plugin_modem_manager.so
