diff --git OLD/gcc.spec NEW/gcc.spec
index ea47834..6d29129 100644
--- OLD/gcc.spec
+++ NEW/gcc.spec
@@ -415,6 +415,7 @@ including templates and exception handling.
 Summary: GNU Standard C++ Library
 Autoreq: true
 Requires: glibc >= 2.10.90-7
+Requires: (libstdc++-pp = %{version}-%{release} if gdb-headless)
 BuildRequires: tzdata >= 2017c
 %if 0%{?fedora} > 38 || 0%{?rhel} > 9
 Recommends: tzdata >= 2017c
@@ -426,6 +427,15 @@ Requires: tzdata >= 2017c
 The libstdc++ package contains a rewritten standard compliant GCC Standard
 C++ Library.
 
+%package -n libstdc++-pp
+Summary: GDB pretty printers for the C++ standard library
+Requires: libstdc++%{?_isa} = %{version}-%{release}
+Autoreq: true
+
+%description -n libstdc++-pp
+This package contains pretty printers for the C++ standard library usable
+from GDB.
+
 %package -n libstdc++-devel
 Summary: Header files and libraries for C++ development
 Requires: libstdc++%{?_isa} = %{version}-%{release}
@@ -3162,7 +3172,9 @@ end
 %doc rpm.doc/changelogs/gcc/cp/ChangeLog*
 
 %files -n libstdc++
-%{_prefix}/%{_lib}/libstdc++.so.6*
+%{_libdir}/libstdc++.so.6*
+
+%files -n libstdc++-pp
 %dir %{_datadir}/gdb
 %dir %{_datadir}/gdb/auto-load
 %dir %{_datadir}/gdb/auto-load/%{_prefix}
@@ -3173,9 +3185,9 @@ end
 %endif
 %{_datadir}/gdb/auto-load/%{_prefix}/%{_lib}/libstdc*gdb.py*
 %{_datadir}/gdb/auto-load/%{_prefix}/%{_lib}/__pycache__
-%dir %{_prefix}/share/gcc-%{gcc_major}
-%dir %{_prefix}/share/gcc-%{gcc_major}/python
-%{_prefix}/share/gcc-%{gcc_major}/python/libstdcxx
+%dir %{_datadir}/gcc-%{gcc_major}
+%dir %{_datadir}/gcc-%{gcc_major}/python
+%{_datadir}/gcc-%{gcc_major}/python/libstdcxx
 
 %files -n libstdc++-devel
 %dir %{_prefix}/include/c++
