--- gdm.spec	2024-11-01 22:58:24.814622618 +0800
+++ gdm.spec.new	2024-11-01 23:15:05.625097364 +0800
@@ -23,12 +23,10 @@
 # Downstream patches
 Patch:          0001-Honor-initial-setup-being-disabled-by-distro-install.patch
 Patch:          0001-data-add-system-dconf-databases-to-gdm-profile.patch
-Patch:          0001-Add-headless-session-files.patch
 
 BuildRequires:  dconf
 BuildRequires:  desktop-file-utils
 BuildRequires:  gettext-devel
-BuildRequires:  libXdmcp-devel
 BuildRequires:  git-core
 BuildRequires:  meson
 BuildRequires:  pam-devel
@@ -63,13 +61,10 @@
 Requires: gnome-settings-daemon >= 3.27.90
 Requires: gnome-shell
 Requires: iso-codes
-# We need 1.0.4-5 since it lets us use "localhost" in auth cookies
-Requires: libXau >= 1.0.4-4
 Requires: pam
 Requires: /sbin/nologin
 Requires: systemd >= 186
 Requires: system-logos
-Requires: python3-pam
 
 # Until the greeter gets dynamic user support, it can't
 # use a user bus
@@ -118,8 +113,8 @@
        -Dplymouth=enabled \
        -Dselinux=enabled\
        -Dwayland-support=true \
-       -Dx11-support=true \
-       -Dxdmcp=enabled
+       -Dx11-support=false \
+       -Dxdmcp=disabled
 
 %meson_build
 
@@ -140,13 +135,9 @@
 # docs go elsewhere
 rm -rf %{buildroot}/%{_prefix}/doc
 
-rm -f %{buildroot}/%{_udevrulesdir}/61-gdm.rules
-
 # create log dir
 mkdir -p %{buildroot}/var/log/gdm
 
-(cd %{buildroot}%{_sysconfdir}/gdm; ln -sf ../X11/xinit/Xsession .)
-
 mkdir -p %{buildroot}%{_datadir}/gdm/autostart/LoginWindow
 
 mkdir -p %{buildroot}/run/gdm
@@ -226,7 +217,6 @@
 %config %{_sysconfdir}/pam.d/gdm-autologin
 %config %{_sysconfdir}/pam.d/gdm-password
 # not config files
-%{_sysconfdir}/gdm/Xsession
 %{_datadir}/gdm/gdm.schemas
 %{_sysconfdir}/dbus-1/system.d/gdm.conf
 %dir %{_sysconfdir}/gdm/Init
@@ -240,7 +230,6 @@
 %{_libexecdir}/gdm-runtime-config
 %{_libexecdir}/gdm-session-worker
 %{_libexecdir}/gdm-wayland-session
-%{_libexecdir}/gdm-headless-login-session
 %{_sbindir}/gdm
 %{_bindir}/gdmflexiserver
 %{_bindir}/gdm-config
@@ -270,13 +259,9 @@
 %config %{_sysconfdir}/pam.d/gdm-fingerprint
 %{_sysconfdir}/pam.d/gdm-launch-environment
 %{_unitdir}/gdm.service
-%{_unitdir}/gnome-headless-session@.service
 %dir %{_userunitdir}/gnome-session@gnome-login.target.d/
 %{_userunitdir}/gnome-session@gnome-login.target.d/session.conf
 %{_sysusersdir}/%{name}.conf
-%{_libexecdir}/gdm-host-chooser
-%{_libexecdir}/gdm-simple-chooser
-%{_libexecdir}/gdm-x-session
 
 %files devel
 %dir %{_includedir}/gdm
