diff --git OLD/gdm.spec NEW/gdm.spec
index e9974ef..2af1fb8 100644
--- OLD/gdm.spec
+++ NEW/gdm.spec
@@ -5,13 +5,8 @@
 %global major_version %%(echo %{version} | cut -d '.' -f1)
 %global tarball_version %%(echo %{version} | tr '~' '.')
 
-# This controls support for launching X11 desktops.
-# gdm itself will always use Wayland.
-%if 0%{?rhel}
+# Disable X11
 %bcond x11 0
-%else
-%bcond x11 1
-%endif
 
 Name:           gdm
 Epoch:          1
@@ -28,7 +23,6 @@ Source2:        gdm.sysusers
 # Downstream patches
 Patch:          0001-Honor-initial-setup-being-disabled-by-distro-install.patch
 Patch:          0001-data-add-system-dconf-databases-to-gdm-profile.patch
-Patch:          0001-Add-headless-session-files.patch
 
 BuildRequires:  dconf
 BuildRequires:  desktop-file-utils
@@ -71,13 +65,14 @@ Requires: gnome-session-wayland-session >= 49~beta
 Requires: gnome-settings-daemon >= 3.27.90
 Requires: gnome-shell
 Requires: iso-codes
+%if %{with x11}
 # We need 1.0.4-5 since it lets us use "localhost" in auth cookies
 Requires: libXau >= 1.0.4-4
+%endif
 Requires: pam
 Requires: /sbin/nologin
 Requires: systemd >= 186
 Requires: system-logos
-Requires: python3-pam
 
 %if %{with x11}
 Requires: xorg-x11-xinit
@@ -89,9 +84,7 @@ Provides: gdm-libs%{?_isa} = %{epoch}:%{version}-%{release}
 GDM, the GNOME Display Manager, handles authentication-related backend
 functionality for logging in a user and unlocking the user's session after
 it's been locked. GDM also provides functionality for initiating user-switching,
-so more than one user can be logged in at the same time. It handles
-graphical session registration with the system for both local and remote
-sessions (in the latter case, via GNOME Remote Desktop and the RDP protocol).
+so more than one user can be logged in at the same time.
 
 %package devel
 Summary: Development files for gdm
@@ -183,13 +176,13 @@ ln -sf ../X11/xinit/Xsession %{buildroot}%{_sysconfdir}/gdm/
 %if %{with x11}
 %{_libexecdir}/gdm-x-session
 %endif
-%{_libexecdir}/gdm-headless-login-session
 %{_sbindir}/gdm
 %{_bindir}/gdmflexiserver
 %{_bindir}/gdm-config
 %dir %{_datadir}/dconf
 %dir %{_datadir}/dconf/profile
 %{_datadir}/dconf/profile/gdm
+%dir %{_datadir}/gdm
 %dir %{_datadir}/gdm/greeter
 %dir %{_datadir}/gdm/greeter/applications
 %{_datadir}/gdm/greeter/applications/*
@@ -200,7 +193,7 @@ ln -sf ../X11/xinit/Xsession %{buildroot}%{_sysconfdir}/gdm/
 %{_datadir}/polkit-1/rules.d/20-gdm.rules
 %{_libdir}/girepository-1.0/Gdm-1.0.typelib
 %{_libdir}/security/pam_gdm.so
-%{_libdir}/libgdm.so.1{,.*}
+%{_libdir}/libgdm.so.*
 %{_libexecdir}/gdm-auth-config-redhat
 %ghost %dir %{_localstatedir}/log/gdm
 %ghost %dir %{_localstatedir}/lib/gdm
@@ -209,7 +202,6 @@ ln -sf ../X11/xinit/Xsession %{buildroot}%{_sysconfdir}/gdm/
 %config %{_sysconfdir}/pam.d/gdm-fingerprint
 %{_sysconfdir}/pam.d/gdm-launch-environment
 %{_unitdir}/gdm.service
-%{_unitdir}/gnome-headless-session@.service
 %dir %{_userunitdir}/gnome-session@gnome-login.target.d/
 %{_userunitdir}/gnome-session@gnome-login.target.d/gnome-login.session.conf
 %{_sysusersdir}/%{name}.conf
