--- gdm.spec	2024-11-01 21:36:57.220979684 +0800
+++ gdm.spec.new	2024-11-01 21:50:22.797262836 +0800
@@ -26,7 +26,6 @@
 Patch:          0001-Honor-initial-setup-being-disabled-by-distro-install.patch
 Patch:          0001-data-add-system-dconf-databases-to-gdm-profile.patch
 Patch:          0001-xorg-detect.patch
-Patch:          0001-Add-headless-session-files.patch
 
 BuildRequires:  dconf
 BuildRequires:  desktop-file-utils
@@ -65,15 +64,10 @@
 Requires: gnome-settings-daemon >= 3.27.90
 Requires: gnome-shell
 Requires: iso-codes
-# We need 1.0.4-5 since it lets us use "localhost" in auth cookies
-Requires: libXau >= 1.0.4-4
 Requires: pam
 Requires: /sbin/nologin
 Requires: systemd >= 186
 Requires: system-logos
-Requires: xhost xmodmap xrdb
-Requires: xorg-x11-xinit
-Requires: python3-pam
 
 # Until the greeter gets dynamic user support, it can't
 # use a user bus
@@ -244,7 +238,6 @@
 %{_libexecdir}/gdm-simple-chooser
 %{_libexecdir}/gdm-wayland-session
 %{_libexecdir}/gdm-x-session
-%{_libexecdir}/gdm-headless-login-session
 %{_sbindir}/gdm
 %{_bindir}/gdmflexiserver
 %{_bindir}/gdm-config
@@ -276,7 +269,6 @@
 %{_sysconfdir}/pam.d/gdm-launch-environment
 %{_udevrulesdir}/61-gdm.rules
 %{_unitdir}/gdm.service
-%{_unitdir}/gnome-headless-session@.service
 %dir %{_userunitdir}/gnome-session@gnome-login.target.d/
 %{_userunitdir}/gnome-session@gnome-login.target.d/session.conf
 %{_sysusersdir}/%{name}.conf
