diff --git OLD/geoclue2.spec NEW/geoclue2.spec
index 183e62c..c99374e 100644
--- OLD/geoclue2.spec
+++ NEW/geoclue2.spec
@@ -79,6 +79,7 @@ The %{name}-demos package contains demo applications that use %{name}.
 
 %conf
 %meson \
+       -Dnmea-source=false \
        -Ddbus-srv-user=geoclue \
        -Ddefault-wifi-url="https://api.beacondb.net/v1/geolocate" \
        -Ddefault-wifi-submit-url="https://api.beacondb.net/v2/geosubmit" \
@@ -115,18 +116,13 @@ exit 0
 %license COPYING
 %doc NEWS
 %config %{_sysconfdir}/geoclue/
-%dir %{_libexecdir}/geoclue-2.0
-%dir %{_libexecdir}/geoclue-2.0/demos
-%{_sysconfdir}/xdg/autostart/geoclue-demo-agent.desktop
 %{_libexecdir}/geoclue
 %{_datadir}/dbus-1/system.d/org.freedesktop.GeoClue2.conf
 %{_datadir}/dbus-1/system.d/org.freedesktop.GeoClue2.Agent.conf
 %{_datadir}/dbus-1/system-services/org.freedesktop.GeoClue2.service
 %{_datadir}/polkit-1/rules.d/org.freedesktop.GeoClue2.rules
-%{_datadir}/applications/geoclue-demo-agent.desktop
 %{_mandir}/man5/geoclue.5*
 %{_unitdir}/geoclue.service
-%{_libexecdir}/geoclue-2.0/demos/agent
 %{_sysusersdir}/geoclue2.conf
 %attr(755,geoclue,geoclue) %dir /var/lib/geoclue
 
@@ -153,7 +149,12 @@ exit 0
 %{_libdir}/libgeoclue-2.so
 
 %files demos
+%dir %{_libexecdir}/geoclue-2.0
+%dir %{_libexecdir}/geoclue-2.0/demos
+%{_sysconfdir}/xdg/autostart/geoclue-demo-agent.desktop
+%{_libexecdir}/geoclue-2.0/demos/agent
 %{_libexecdir}/geoclue-2.0/demos/where-am-i
+%{_datadir}/applications/geoclue-demo-agent.desktop
 %{_datadir}/applications/geoclue-where-am-i.desktop
 
 
