diff --git OLD/gnome-shell.spec NEW/gnome-shell.spec
index 76bae7b..c34db85 100644
--- OLD/gnome-shell.spec
+++ NEW/gnome-shell.spec
@@ -180,6 +180,14 @@ BuildArch: noarch
 %description common
 %{summary}
 
+%package devel
+Summary: Development files for %{name}
+Requires: %{name} = %{version}-%{release}
+
+%description devel
+%{summary}
+This package contains development files for %{name}.
+
 %prep
 %autosetup -S git -n %{name}-%{tarball_version}
 
@@ -224,6 +232,7 @@ desktop-file-validate %{buildroot}%{_datadir}/applications/org.gnome.Shell.Porta
 %{_datadir}/gnome-control-center/keybindings/50-gnome-shell-screenshots.xml
 %{_datadir}/gnome-control-center/keybindings/50-gnome-shell-system.xml
 %{_datadir}/gnome-shell/
+%exclude %{_datadir}/gnome-shell/*.gir
 %{_datadir}/dbus-1/services/org.gnome.ScreenSaver.service
 %{_datadir}/dbus-1/services/org.gnome.Shell.CalendarServer.service
 %{_datadir}/dbus-1/services/org.gnome.Shell.Extensions.service
@@ -264,5 +273,8 @@ desktop-file-validate %{buildroot}%{_datadir}/applications/org.gnome.Shell.Porta
 %files common
 %{_datadir}/glib-2.0/schemas/*.xml
 
+%files devel
+%{_datadir}/gnome-shell/*.gir
+
 %changelog
 %autochangelog
