--- gnome-keyring.spec	2025-02-05 19:15:46.337141166 +0800
+++ gnome-keyring.spec.new	2025-02-05 19:17:32.945051525 +0800
@@ -2,7 +2,8 @@
 %global gcr_version 3.27.90
 %global gcrypt_version 1.2.2
 
-%bcond_without ssh_agent
+# Disable ssh_agent for all builds
+%bcond ssh_agent 0
 
 Name:           gnome-keyring
 Version:        46.2
@@ -40,8 +41,7 @@
 Requires: /usr/bin/ssh-add
 Requires: /usr/bin/ssh-agent
 %endif
-# for /usr/libexec/gcr-ssh-askpass
-Requires: gcr3
+Requires: %{_libexecdir}/gcr-ssh-askpass
 
 %description
 The gnome-keyring session daemon manages passwords and other types of
