diff --git OLD/gnome-control-center.spec NEW/gnome-control-center.spec
index 8c6dee0..4d39225 100644
--- OLD/gnome-control-center.spec
+++ NEW/gnome-control-center.spec
@@ -12,8 +12,8 @@
 
 %global tarball_version %%(echo %{version} | tr '~' '.')
 
-# Disable parental control for RHEL builds
-%bcond malcontent %[!0%{?rhel}]
+# Disable malcontent for all builds
+%bcond malcontent 0
 
 Name:           gnome-control-center
 Version:        48.1
