diff --git OLD/gnome-software.spec NEW/gnome-software.spec
index 90caad9..e64fc6a 100644
--- OLD/gnome-software.spec
+++ NEW/gnome-software.spec
@@ -11,12 +11,12 @@
 
 # Disable WebApps for RHEL builds
 %bcond webapps %[!0%{?rhel}]
-# Disable parental control for RHEL builds
-%bcond malcontent %[!0%{?rhel}]
+# Disable malcontent for all builds
+%bcond malcontent 0
 # Disable rpm-ostree support for RHEL builds
 %bcond rpmostree %[!0%{?rhel}]
-# Disable DKMS/akmods support for RHEL builds
-%bcond dkms %[!0%{?rhel}]
+# Disable DKMS/akmods support for all builds
+%bcond dkms 0
 
 %bcond packagekit 0
 %bcond dnf5 1
