diff --git OLD/gnome-session.spec NEW/gnome-session.spec
index c524be0..091b749 100644
--- OLD/gnome-session.spec
+++ NEW/gnome-session.spec
@@ -49,7 +49,6 @@ the other core GNOME components and handles logout and saving the session.
 %package wayland-session
 Summary: Desktop file for wayland based gnome session
 Requires: %{name}%{?_isa} = %{version}-%{release}
-Requires: xorg-x11-server-Xwayland%{?_isa} >= 1.20.99.1
 Requires: gnome-shell
 Obsoletes: gnome-session-xsession < %{version}-%{release}
 
