diff --git OLD/gnome-shell.spec NEW/gnome-shell.spec
index 76bae7b..beae12d 100644
--- OLD/gnome-shell.spec
+++ NEW/gnome-shell.spec
@@ -116,8 +116,6 @@ Requires:       accountsservice-libs%{?_isa}
 Requires:       gdm-libs%{?_isa}
 # needed for settings items in menus
 Requires:       gnome-control-center
-# needed by some utilities
-Requires:       python3%{_isa}
 # needed for the dual-GPU launch menu
 Requires:       switcheroo-control
 # needed for clocks/weather integration
@@ -180,6 +178,14 @@ BuildArch: noarch
 %description common
 %{summary}
 
+%package devel
+Summary: Development files for %{name}
+Requires: %{name} = %{version}-%{release}
+
+%description devel
+%{summary}
+This package contains development files for %{name}.
+
 %prep
 %autosetup -S git -n %{name}-%{tarball_version}
 
@@ -201,6 +207,9 @@ BuildArch: noarch
 mkdir -p %{buildroot}%{_datadir}/gnome-shell/extensions
 mkdir -p %{buildroot}%{_datadir}/gnome-shell/search-providers
 
+rm %{buildroot}%{_bindir}/gnome-shell-extension-tool
+rm %{buildroot}%{_bindir}/gnome-shell-test-tool
+
 %find_lang %{name}
 
 %check
@@ -215,8 +224,6 @@ desktop-file-validate %{buildroot}%{_datadir}/applications/org.gnome.Shell.Porta
 %doc NEWS README.md
 %{_bindir}/gnome-shell
 %{_bindir}/gnome-extensions
-%{_bindir}/gnome-shell-extension-tool
-%{_bindir}/gnome-shell-test-tool
 %{_datadir}/glib-2.0/schemas/00_org.gnome.shell.gschema.override
 %{_datadir}/applications/org.gnome.Shell.Extensions.desktop
 %{_datadir}/bash-completion/completions/gnome-extensions
@@ -224,6 +231,7 @@ desktop-file-validate %{buildroot}%{_datadir}/applications/org.gnome.Shell.Porta
 %{_datadir}/gnome-control-center/keybindings/50-gnome-shell-screenshots.xml
 %{_datadir}/gnome-control-center/keybindings/50-gnome-shell-system.xml
 %{_datadir}/gnome-shell/
+%exclude %{_datadir}/gnome-shell/*.gir
 %{_datadir}/dbus-1/services/org.gnome.ScreenSaver.service
 %{_datadir}/dbus-1/services/org.gnome.Shell.CalendarServer.service
 %{_datadir}/dbus-1/services/org.gnome.Shell.Extensions.service
@@ -264,5 +272,8 @@ desktop-file-validate %{buildroot}%{_datadir}/applications/org.gnome.Shell.Porta
 %files common
 %{_datadir}/glib-2.0/schemas/*.xml
 
+%files devel
+%{_datadir}/gnome-shell/*.gir
+
 %changelog
 %autochangelog
