diff --git OLD/gnupg2.spec NEW/gnupg2.spec
index 7832d4d..fda1662 100644
--- OLD/gnupg2.spec
+++ NEW/gnupg2.spec
@@ -1,4 +1,5 @@
 %bcond_with bootstrap
+%bcond_with ldap
 
 %global split_min_version 2.4.9-4.fc42.1
 
@@ -65,7 +66,9 @@ BuildRequires: libassuan-devel >= 2.5.0
 BuildRequires: libgcrypt-devel >= 1.9.1
 BuildRequires: libgpg-error-devel >= 1.46
 BuildRequires: libksba-devel >= 1.6.3
+%if %{with ldap}
 BuildRequires: openldap-devel
+%endif
 BuildRequires: pcsc-lite-libs
 BuildRequires: ncurses-devel
 BuildRequires: npth-devel
@@ -85,11 +88,11 @@ BuildRequires: swtpm
 Requires: libgcrypt >= 1.9.1
 Requires: libgpg-error >= 1.46
 
-Requires: gnupg2-dirmngr%{?_isa} = %{version}-%{release}
-Requires: gnupg2-gpgconf%{?_isa} = %{version}-%{release}
-Requires: gnupg2-gpg-agent%{?_isa} = %{version}-%{release}
+Recommends: gnupg2-dirmngr%{?_isa} = %{version}-%{release}
+Recommends: gnupg2-gpgconf%{?_isa} = %{version}-%{release}
+Recommends: gnupg2-gpg-agent%{?_isa} = %{version}-%{release}
 Requires: gnupg2-keyboxd%{?_isa} = %{version}-%{release}
-Requires: gnupg2-verify%{?_isa} = %{version}-%{release}
+Recommends: gnupg2-verify%{?_isa} = %{version}-%{release}
 
 Recommends: pinentry
 
@@ -257,7 +260,13 @@ sed -i -e 's/"libpcsclite\.so"/"%{pcsclib}"/' scd/scdaemon.c
   --enable-g13 \
   --disable-ccid-driver \
   --with-tss=intel \
-  --enable-large-secmem
+  --enable-large-secmem \
+%if %{with ldap}
+  --enable-ldap \
+%else
+  --disable-ldap \
+%endif
+  %{nil}
 
 # need scratch gpg database for tests
 mkdir -p $HOME/.gnupg
@@ -281,6 +290,8 @@ install -p %{SOURCE3} %{SOURCE4} %{buildroot}%{_sysconfdir}/profile.d
 install -m644 -p AUTHORS NEWS THANKS TODO \
   %{buildroot}%{_pkgdocdir}
 
+mv %{buildroot}%{_datadir}/gnupg/help*.txt %{buildroot}%{_pkgdocdir}/
+
 # compat symlinks
 ln -sf gpg %{buildroot}%{_bindir}/gpg2
 ln -sf gpgv %{buildroot}%{_bindir}/gpgv2
@@ -346,13 +357,14 @@ make -k check
 %{_mandir}/man?/gnupg.*
 %{_mandir}/man?/gnupg2.*
 %{_pkgdocdir}/
-%{_datadir}/gnupg/help*.txt
 
 %files dirmngr
 %license COPYING
 %{_bindir}/dirmngr
 %{_bindir}/dirmngr-client
+%if %{with ldap}
 %{_libexecdir}/dirmngr_ldap
+%endif
 %{_userunitdir}/dirmngr.*
 %{_mandir}/man?/dirmngr.*
 %{_mandir}/man?/dirmngr-client.*
