diff --git OLD/gstreamer1-plugins-base.spec NEW/gstreamer1-plugins-base.spec
index eafca42..5b1669c 100644
--- OLD/gstreamer1-plugins-base.spec
+++ NEW/gstreamer1-plugins-base.spec
@@ -1,5 +1,6 @@
-%bcond cdparanoia %{undefined rhel}
-%bcond libvisual %{undefined rhel}
+# Disable cdparanoia and libvisual for all builds
+%bcond cdparanoia 0
+%bcond libvisual 0
 
 %global         majorminor      1.0
 
