diff --git OLD/gvfs.spec NEW/gvfs.spec
index 39d05d2..2fc0226 100644
--- OLD/gvfs.spec
+++ NEW/gvfs.spec
@@ -19,6 +19,9 @@
 %global talloc_version 1.3.0
 %global udisks2_version 1.97
 
+%bcond dnssd 0
+%bcond bluray 0
+
 Name:    gvfs
 Version: 1.60.0
 Release: %autorelease
@@ -35,20 +38,23 @@ BuildRequires: pkgconfig(glib-2.0) >= %{glib2_version}
 BuildRequires: pkgconfig(dbus-1)
 BuildRequires: pkgconfig(gcr-4)
 BuildRequires: pkgconfig(gsettings-desktop-schemas) >= %{gsettings_desktop_schemas_version}
-BuildRequires: /usr/bin/ssh
 %if ! (0%{?rhel} >= 10)
 BuildRequires: pkgconfig(libcdio_paranoia) >= %{libcdio_paranoia_version}
 %endif
 BuildRequires: pkgconfig(gudev-1.0) >= %{gudev_version}
 BuildRequires: pkgconfig(libsoup-3.0) >= %{libsoup_version}
+%if %{with dnssd}
 BuildRequires: pkgconfig(avahi-client) >= %{avahi_version}
 BuildRequires: pkgconfig(avahi-glib) >= %{avahi_version}
+%endif
 BuildRequires: pkgconfig(libsecret-1)
 BuildRequires: gettext-devel
 BuildRequires: pkgconfig(udisks2) >= %{udisks2_version}
+%if %{with bluray}
 %if ! 0%{?rhel}
 BuildRequires: pkgconfig(libbluray)
 %endif
+%endif
 BuildRequires: systemd-devel >= %{systemd_version}
 BuildRequires: pkgconfig(libxslt)
 BuildRequires: docbook-style-xsl
@@ -59,8 +65,8 @@ Requires: %{name}-client%{?_isa} = %{version}-%{release}
 Requires: glib2%{?_isa} >= %{glib2_version}
 Requires: gsettings-desktop-schemas >= %{gsettings_desktop_schemas_version}
 Requires: polkit
-Requires: udisks2 >= %{udisks2_version}
-Requires: /usr/bin/wsdd
+
+Requires: (%{name}-udisks2%{?_isa} = %{version}-%{release} if udev)
 
 Obsoletes: gnome-mount <= 0.8
 Obsoletes: gnome-mount-nautilus-properties <= 0.8
@@ -74,7 +80,7 @@ Obsoletes: gvfs-mtp < 1.55.1-4
 
 %description
 The gvfs package provides backend implementations for the gio
-framework in GLib. It includes ftp, sftp, cifs.
+framework in GLib.
 
 
 %package client
@@ -86,6 +92,59 @@ The gvfs package provides client modules of backend implementations for the gio
 framework in GLib.
 
 
+%package udisks2
+Summary: UDisks2 support for gvfs
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: udisks2 >= %{udisks2_version}
+
+%description udisks2
+This package provides support for accessing files on UDisks2 volumes.
+
+
+%package misc
+Summary: Miscellaneous modules of backends for the gio framework in GLib
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: %{name}-client%{?_isa} = %{version}-%{release}
+%if ! (0%{?rhel} >= 10)
+BuildRequires: pkgconfig(libcdio_paranoia) >= %{libcdio_paranoia_version}
+%endif
+
+%description misc
+The gvfs package provides misc modules of backend implementations for the gio
+framework in GLib.
+
+
+%package network
+Summary: Network support for gvfs
+BuildRequires: /usr/bin/ssh
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: %{name}-client%{?_isa} = %{version}-%{release}
+
+%description network
+This package provides support for accessing files on remote servers
+to applications using gvfs.
+
+
+%package webdav
+Summary: WebDAV support for gvfs
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: %{name}-client%{?_isa} = %{version}-%{release}
+
+%description webdav
+This package provides support for accessing files on WebDAV servers
+to applications using gvfs.
+
+
+%package wsdd
+Summary: WSDD support for gvfs
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: %{name}-client%{?_isa} = %{version}-%{release}
+Requires: /usr/bin/wsdd
+
+%description wsdd
+This package provides support for wsdd discovery to applications using gvfs.
+
+
 %package fuse
 Summary: FUSE support for gvfs
 Requires: %{name}%{?_isa} = %{version}-%{release}
@@ -102,6 +161,7 @@ to access the gvfs filesystems.
 Summary: Windows fileshare support for gvfs
 Requires: %{name}%{?_isa} = %{version}-%{release}
 Requires: %{name}-client%{?_isa} = %{version}-%{release}
+Requires: %{name}-wsdd%{?_isa} = %{version}-%{release}
 BuildRequires: libsmbclient-devel >= %{libsmbclient_version}
 BuildRequires: pkgconfig(talloc) >= %{talloc_version}
 
@@ -235,6 +295,16 @@ file services.
 %endif
 %if 0%{?rhel} >= 10
        -Dcdda=false \
+%endif
+%if %{with dnssd}
+       -Ddnssd=true \
+%else
+       -Ddnssd=false \
+%endif
+%if %{with bluray}
+       -Dbluray=true \
+%else
+       -Dbluray=false \
 %endif
         %{nil}
 %meson_build
@@ -252,6 +322,16 @@ cp -p daemon/trashlib/COPYING COPYING.GPL3
 killall -USR1 gvfsd >&/dev/null || :
 
 # Reload .mount files when single subpackage is installed:
+%post udisks2
+killall -USR1 gvfsd >&/dev/null || :
+%post misc
+killall -USR1 gvfsd >&/dev/null || :
+%post network
+killall -USR1 gvfsd >&/dev/null || :
+%post webdav
+killall -USR1 gvfsd >&/dev/null || :
+%post wsdd
+killall -USR1 gvfsd >&/dev/null || :
 %post smb
 killall -USR1 gvfsd >&/dev/null || :
 %ifnarch s390 s390x
@@ -287,56 +367,34 @@ killall -USR1 gvfsd >&/dev/null || :
 %dir %{_datadir}/gvfs
 %dir %{_datadir}/gvfs/mounts
 %{_datadir}/gvfs/mounts/admin.mount
-%{_datadir}/gvfs/mounts/sftp.mount
 %{_datadir}/gvfs/mounts/trash.mount
-%if ! (0%{?rhel} >= 10)
-%{_datadir}/gvfs/mounts/cdda.mount
-%endif
 %{_datadir}/gvfs/mounts/computer.mount
-%{_datadir}/gvfs/mounts/dav.mount
-%{_datadir}/gvfs/mounts/dav+sd.mount
-%{_datadir}/gvfs/mounts/http.mount
-%{_datadir}/gvfs/mounts/localtest.mount
+%if %{with dnssd}
 %{_datadir}/gvfs/mounts/dns-sd.mount
+%endif
 %{_datadir}/gvfs/mounts/network.mount
-%{_datadir}/gvfs/mounts/ftp.mount
-%{_datadir}/gvfs/mounts/ftpis.mount
-%{_datadir}/gvfs/mounts/ftps.mount
 %{_datadir}/gvfs/mounts/recent.mount
-%{_datadir}/gvfs/mounts/wsdd.mount
 %{_datadir}/dbus-1/services/org.gtk.vfs.Daemon.service
 %{_datadir}/dbus-1/services/org.gtk.vfs.Metadata.service
-%{_datadir}/dbus-1/services/org.gtk.vfs.UDisks2VolumeMonitor.service
 %dir %{_datadir}/gvfs/remote-volume-monitors
-%{_datadir}/gvfs/remote-volume-monitors/udisks2.monitor
-%{_datadir}/GConf/gsettings/*.convert
-%{_datadir}/glib-2.0/schemas/*.xml
+%{_datadir}/glib-2.0/schemas/org.gnome.system.gvfs.enums.xml
 %{_datadir}/polkit-1/actions/org.gtk.vfs.file-operations.policy
 %{_datadir}/polkit-1/rules.d/org.gtk.vfs.file-operations.rules
 %{_libdir}/gvfs/libgvfsdaemon.so
 %{_libexecdir}/gvfsd
 %{_libexecdir}/gvfsd-admin
-%{_libexecdir}/gvfsd-ftp
-%{_libexecdir}/gvfsd-sftp
 %{_libexecdir}/gvfsd-trash
-%if ! (0%{?rhel} >= 10)
-%{_libexecdir}/gvfsd-cdda
-%endif
 %{_libexecdir}/gvfsd-computer
-%{_libexecdir}/gvfsd-dav
-%{_libexecdir}/gvfsd-http
-%{_libexecdir}/gvfsd-localtest
+%if %{with dnssd}
 %{_libexecdir}/gvfsd-dnssd
+%endif
 %{_libexecdir}/gvfsd-network
 %{_libexecdir}/gvfsd-metadata
-%{_libexecdir}/gvfs-udisks2-volume-monitor
 %{_libexecdir}/gvfsd-recent
-%{_libexecdir}/gvfsd-wsdd
 %{_mandir}/man1/gvfsd.1*
 %{_mandir}/man1/gvfsd-metadata.1*
 %{_userunitdir}/gvfs-daemon.service
 %{_userunitdir}/gvfs-metadata.service
-%{_userunitdir}/gvfs-udisks2-volume-monitor.service
 
 %files client -f gvfs.lang
 %license COPYING COPYING.GPL3
@@ -347,6 +405,44 @@ killall -USR1 gvfsd >&/dev/null || :
 %{_libdir}/gio/modules/libgvfsdbus.so
 %{_mandir}/man7/gvfs.7*
 
+%files udisks2
+%{_libexecdir}/gvfs-udisks2-volume-monitor
+%{_datadir}/dbus-1/services/org.gtk.vfs.UDisks2VolumeMonitor.service
+%{_datadir}/gvfs/remote-volume-monitors/udisks2.monitor
+%{_userunitdir}/gvfs-udisks2-volume-monitor.service
+
+%files misc
+%if ! (0%{?rhel} >= 10)
+%{_libexecdir}/gvfsd-cdda
+%endif
+%{_libexecdir}/gvfsd-localtest
+%if ! (0%{?rhel} >= 10)
+%{_datadir}/gvfs/mounts/cdda.mount
+%endif
+%{_datadir}/gvfs/mounts/localtest.mount
+
+%files network
+%{_libexecdir}/gvfsd-ftp
+%{_libexecdir}/gvfsd-sftp
+%{_libexecdir}/gvfsd-http
+%{_datadir}/gvfs/mounts/sftp.mount
+%{_datadir}/gvfs/mounts/http.mount
+%{_datadir}/gvfs/mounts/ftp.mount
+%{_datadir}/gvfs/mounts/ftpis.mount
+%{_datadir}/gvfs/mounts/ftps.mount
+
+%files webdav
+%{_libexecdir}/gvfsd-dav
+%{_datadir}/gvfs/mounts/dav.mount
+%if %{with dnssd}
+%{_datadir}/gvfs/mounts/dav+sd.mount
+%endif
+
+%files wsdd
+%{_libexecdir}/gvfsd-wsdd
+%{_datadir}/gvfs/mounts/wsdd.mount
+%{_datadir}/glib-2.0/schemas/org.gnome.system.wsdd.gschema.xml
+
 %files fuse
 %{_libexecdir}/gvfsd-fuse
 %{_mandir}/man1/gvfsd-fuse.1*
@@ -357,6 +453,8 @@ killall -USR1 gvfsd >&/dev/null || :
 %{_libexecdir}/gvfsd-smb-browse
 %{_datadir}/gvfs/mounts/smb-browse.mount
 %{_datadir}/gvfs/mounts/smb.mount
+%{_datadir}/GConf/gsettings/gvfs-smb.convert
+%{_datadir}/glib-2.0/schemas/org.gnome.system.smb.gschema.xml
 
 
 %if ! (0%{?rhel} >= 9)
