diff --git OLD/ibus.spec NEW/ibus.spec
index 0db24df..4929491 100644
--- OLD/ibus.spec
+++ NEW/ibus.spec
@@ -1,19 +1,9 @@
-%global source_version %%(echo "%version" | tr '~' '-')
-
-# https://fedoraproject.org/wiki/Changes/No_more_automagic_Python_bytecompilation_phase_3
-%if (0%{?fedora} > 29 || 0%{?rhel} > 7)
-%global with_python2 0
-%else
-%global with_python2 1
-%endif
+%global source_version %%(echo "%{version}" | tr '~' '-')
 
 %global with_pkg_config %(pkg-config --version >/dev/null 2>&1 && echo -n "1" || echo -n "0")
 
 %global ibus_api_version 1.0
-%global pkgcache /var/cache/%name
-
-# for bytecompile in %%{_datadir}/ibus/setup
-%global __python %{__python3}
+%global pkgcache /var/cache/%{name}
 
 # No gtk2 in RHEL 10
 %if 0%{?rhel} > 9
@@ -66,9 +56,9 @@ Version:        1.5.33
 Release:        3%{?dist}
 Summary:        Intelligent Input Bus for Linux OS
 License:        LGPL-2.1-or-later
-URL:            https://github.com/ibus/%name/wiki
-Source0:        https://github.com/ibus/%name/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz
-Source1:        https://github.com/ibus/%name/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz.sum#/%{name}.tar.gz.sum
+URL:            https://github.com/ibus/%{name}/wiki
+Source0:        https://github.com/ibus/%{name}/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz
+Source1:        https://github.com/ibus/%{name}/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz.sum#/%{name}.tar.gz.sum
 Source2:        %{name}-xinput
 Source3:        %{name}.conf.5
 # Patch0:         %%{name}-HEAD.patch
@@ -94,15 +84,6 @@ BuildRequires:  gtk-doc
 BuildRequires:  dconf-devel
 BuildRequires:  dbus-x11
 BuildRequires:  python3-devel
-BuildRequires:  python3-gobject
-%if %with_python2
-# https://bugzilla.gnome.org/show_bug.cgi?id=759334
-# Need python2 for gsettings-schema-convert
-BuildRequires:  python2-devel
-# for AM_GCONF_SOURCE_2 in configure.ac
-BuildRequires:  GConf2-devel
-BuildRequires:  intltool
-%endif
 BuildRequires:  git
 BuildRequires:  vala
 BuildRequires:  iso-codes-devel
@@ -118,8 +99,8 @@ Requires:       %{name}-libs%{?_isa}   = %{version}-%{release}
 %if %{with gtk2}
 Requires:      (%{name}-gtk2%{?_isa}   = %{version}-%{release} if gtk2)
 %endif
-Requires:       %{name}-gtk3%{?_isa}   = %{version}-%{release}
-Requires:       %{name}-setup          = %{version}-%{release}
+Requires:      (%{name}-gtk3%{?_isa}   = %{version}-%{release} if gtk3)
+Recommends:     %{name}-setup          = %{version}-%{release}
 %if 0%{?fedora}
 Requires:      (%{name}-panel%{?_isa}  = %{version}-%{release} if %ibus_panel_condition)
 %endif
@@ -128,17 +109,9 @@ Requires:      (%{name}-xinit          = %{version}-%{release} if %ibus_xinit_co
 %endif
 
 Requires:       iso-codes
-Requires:       dconf
 # rpmlint asks to delete librsvg2
 #Requires:       librsvg2
-# Owner of %%python3_sitearch/gi/overrides
-Requires:       python3-gobject
-# https://bugzilla.redhat.com/show_bug.cgi?id=1161871
-%{?__python3:Requires: %{__python3}}
 
-Requires:               desktop-file-utils
-Requires(post):         desktop-file-utils
-Requires(postun):       desktop-file-utils
 Requires:               dconf
 Requires(postun):       dconf
 Requires(posttrans):    dconf
@@ -152,6 +125,16 @@ Requires(postun):       %{_sbindir}/alternatives
 %description
 IBus means Intelligent Input Bus. It is an input framework for Linux OS.
 
+%package -n python3-ibus
+Summary:        Python 3 GObject Introspection overrides for IBus
+Requires:       %{name}%{?_isa} = %{version}-%{release}
+Requires:       %{py3_dist pygobject}
+
+%description -n python3-ibus
+The python3-ibus package provides GObject Introspection overrides
+for IBus, allowing Python applications to use the IBus library
+for input method support.
+
 %package libs
 Summary:        IBus libraries
 
@@ -204,8 +187,8 @@ This package contains IBus IM module for GTK4
 %package setup
 Summary:        IBus setup utility
 Requires:       %{name} = %{version}-%{release}
-%{?__python3:Requires: %{__python3}}
 Requires:       python3-gobject
+Requires:       python3-ibus
 BuildRequires:  gobject-introspection-devel
 BuildRequires:  python3-gobject-devel
 BuildRequires:  make
@@ -214,38 +197,6 @@ BuildArch:      noarch
 %description setup
 This is a setup utility for IBus.
 
-%if %with_python2
-%package pygtk2
-Summary:        IBus PyGTK2 library
-%if (0%{?fedora} && 0%{?fedora} <= 27) || (0%{?rhel} && 0%{?rhel} <= 7)
-Requires:       dbus-python >= %{dbus_python_version}
-%else
-Requires:       python2-dbus >= %{dbus_python_version}
-%endif
-Requires:       python2
-Requires:       pygtk2
-BuildArch:      noarch
-
-%description pygtk2
-This is a PyGTK2 library for IBus. Now major IBus engines use PyGObject3
-and this package will be deprecated.
-%endif
-
-%package py2override
-Summary:        IBus Python2 override library
-Requires:       %{name}-libs%{?_isa}   = %{version}-%{release}
-# Owner of %%python2_sitearch/gi/overrides
-%if (0%{?fedora} && 0%{?fedora} <= 27) || (0%{?rhel} && 0%{?rhel} <= 7)
-Requires:       pygobject3-base
-%else
-Requires:       python2-gobject-base
-%endif
-Requires:       python2
-
-%description py2override
-This is a Python2 override library for IBus. The Python files override
-some functions in GObject-Introspection.
-
 %package wayland
 Summary:        IBus IM module for Wayland
 Requires:       %{name}-libs%{?_isa}   = %{version}-%{release}
@@ -328,8 +279,8 @@ the functionality of the installed %{name} package.
 
 
 %prep
-SAVED_SUM=$(grep sha512sum %SOURCE1 | awk '{print $2}')
-MY_SUM=$(sha512sum %SOURCE0 | awk '{print $1}')
+SAVED_SUM=$(grep sha512sum %{SOURCE1} | awk '{print $2}')
+MY_SUM=$(sha512sum %{SOURCE0} | awk '{print $1}')
 if test x"$SAVED_SUM" != x"$MY_SUM" ; then
     abort
 fi
@@ -375,11 +326,7 @@ fi
     --enable-gtk-doc \
     --enable-surrounding-text \
     --with-python=python3 \
-%if ! %with_python2
     --disable-python2 \
-%else
-    --enable-python-library \
-%endif
     --with-python-overrides-dir=%{python3_sitearch}/gi/overrides \
     --enable-wayland \
     --enable-introspection \
@@ -417,18 +364,14 @@ done
 # install xinput config file
 install -pm 644 -D %{SOURCE2} $RPM_BUILD_ROOT%{_xinputconf}
 
-install -m 755 -d $RPM_BUILD_ROOT%pkgcache/bus
+install -m 755 -d $RPM_BUILD_ROOT%{pkgcache}/bus
 # `rpm -Vaq ibus` compare st_mode of struct stat with lstat(2) and
 # st_mode of the RPM cache and if the file does not exist, st_mode of
 # RPM cache is o0100000 while the actual st_mode is o0100644.
-touch $RPM_BUILD_ROOT%pkgcache/bus/registry
+touch $RPM_BUILD_ROOT%{pkgcache}/bus/registry
 
 # install .desktop files
-%if %with_python2
-echo "NoDisplay=true" >> $RPM_BUILD_ROOT%{_datadir}/applications/ibus-setup.desktop
-%else
 echo "NoDisplay=true" >> $RPM_BUILD_ROOT%{_datadir}/applications/org.freedesktop.IBus.Setup.desktop
-%endif
 #echo "X-GNOME-Autostart-enabled=false" >> $RPM_BUILD_ROOT%%{_sysconfdir}/xdg/autostart/ibus.desktop
 
 mkdir -p $RPM_BUILD_ROOT%{_libdir}/ibus
@@ -521,7 +464,6 @@ dconf update || :
 %{_datadir}/ibus/dicts
 %dir %{_datadir}/ibus/engine
 %{_datadir}/ibus/keymaps
-%{_datadir}/icons/hicolor/*/apps/*
 %{_datadir}/man/man1/ibus.1.gz
 %{_datadir}/man/man1/ibus-daemon.1.gz
 %{_datadir}/man/man7/ibus-emoji.7.gz
@@ -540,12 +482,14 @@ dconf update || :
 %dir %{_prefix}/lib/systemd/user/gnome-session.target.wants
 %{_prefix}/lib/systemd/user/gnome-session.target.wants/*.service
 %{_prefix}/lib/systemd/user/org.freedesktop.IBus.session.*.service
-%python3_sitearch/gi/overrides/__pycache__/*.py*
-%python3_sitearch/gi/overrides/IBus.py
-%verify(not mtime) %dir %pkgcache
-%verify(not mtime) %dir %pkgcache/bus
+%verify(not mtime) %dir %{pkgcache}
+%verify(not mtime) %dir %{pkgcache}/bus
 # 'ibus write-cache --system' updates the system cache.
-%ghost %pkgcache/bus/registry
+%ghost %{pkgcache}/bus/registry
+
+%files -n python3-ibus
+%{python3_sitearch}/gi/overrides/__pycache__/*.py*
+%{python3_sitearch}/gi/overrides/IBus.py
 
 %files libs
 %{_libdir}/libibus-*%{ibus_api_version}.so.*
@@ -566,29 +510,13 @@ dconf update || :
 %{_libdir}/gtk-4.0/%{gtk4_binary_version}/immodules/libim-ibus.so
 %endif
 
-# The setup package won't include icon files so that
-# gtk-update-icon-cache is executed in the main package only one time.
 %files setup
 %{_bindir}/ibus-setup
-%if %with_python2
-%{_datadir}/applications/ibus-setup.desktop
-%else
 %{_datadir}/applications/org.freedesktop.IBus.Setup.desktop
-%endif
 %{_datadir}/ibus/setup
+%{_datadir}/icons/hicolor/*/apps/*
 %{_datadir}/man/man1/ibus-setup.1.gz
 
-%if %with_python2
-%files pygtk2
-%dir %{python2_sitelib}/ibus
-%{python2_sitelib}/ibus/*
-%endif
-
-%if %with_python2
-%files py2override
-%python2_sitearch/gi/overrides/IBus.py*
-%endif
-
 %files wayland
 %{_libexecdir}/ibus-wayland
 
