diff --git OLD/ibus.spec NEW/ibus.spec
index a028cea..9df99ce 100644
--- OLD/ibus.spec
+++ NEW/ibus.spec
@@ -1,9 +1,9 @@
-%global source_version %%(echo "%version" | tr '~' '-')
+%global source_version %%(echo "%{version}" | tr '~' '-')
 
 %global with_pkg_config %(pkg-config --version >/dev/null 2>&1 && echo -n "1" || echo -n "0")
 
 %global ibus_api_version 1.0
-%global pkgcache /var/cache/%name
+%global pkgcache /var/cache/%{name}
 
 # No gtk2 in RHEL 10
 %if 0%{?rhel} > 9
@@ -61,9 +61,9 @@ Version:        1.5.34~rc2
 Release:        1%{?dist}
 Summary:        Intelligent Input Bus for Linux OS
 License:        LGPL-2.1-or-later
-URL:            https://github.com/ibus/%name/wiki
-Source0:        https://github.com/ibus/%name/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz
-Source1:        https://github.com/ibus/%name/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz.sum#/%{name}.tar.gz.sum
+URL:            https://github.com/ibus/%{name}/wiki
+Source0:        https://github.com/ibus/%{name}/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz
+Source1:        https://github.com/ibus/%{name}/releases/download/%{source_version}/%{name}-%{source_version}.tar.gz.sum#/%{name}.tar.gz.sum
 Source2:        %{name}-xinput
 Source3:        %{name}.conf.5
 # Patch0:         %%{name}-HEAD.patch
@@ -104,14 +104,13 @@ Requires:       %{name}-libs%{?_isa}   = %{version}-%{release}
 Requires:      (%{name}-gtk2%{?_isa}   = %{version}-%{release} if gtk2)
 %endif
 Requires:      (%{name}-gtk3%{?_isa}   = %{version}-%{release} if gtk3)
+Recommends:     %{name}-setup          = %{version}-%{release}
 %if 0%{?fedora}
 Requires:      (%{name}-panel%{?_isa}  = %{version}-%{release} if %ibus_panel_condition)
 %endif
 %if %{with xinit}
 Requires:      (%{name}-xinit          = %{version}-%{release} if %ibus_xinit_condition)
 %endif
-Requires:       python3-ibus           = %{version}-%{release}
-Recommends:     %{name}-setup          = %{version}-%{release}
 
 Requires:       iso-codes
 # rpmlint asks to delete librsvg2
@@ -193,6 +192,7 @@ This package contains IBus IM module for GTK4
 Summary:        IBus setup utility
 Requires:       %{name} = %{version}-%{release}
 Requires:       python3-gobject
+Requires:       python3-ibus
 BuildRequires:  gobject-introspection-devel
 BuildRequires:  python3-gobject-devel
 BuildRequires:  make
@@ -283,8 +283,8 @@ the functionality of the installed %{name} package.
 
 
 %prep
-SAVED_SUM=$(grep sha512sum %SOURCE1 | awk '{print $2}')
-MY_SUM=$(sha512sum %SOURCE0 | awk '{print $1}')
+SAVED_SUM=$(grep sha512sum %{SOURCE1} | awk '{print $2}')
+MY_SUM=$(sha512sum %{SOURCE0} | awk '{print $1}')
 if test x"$SAVED_SUM" != x"$MY_SUM" ; then
     abort
 fi
@@ -368,11 +368,11 @@ done
 # install xinput config file
 install -pm 644 -D %{SOURCE2} $RPM_BUILD_ROOT%{_xinputconf}
 
-install -m 755 -d $RPM_BUILD_ROOT%pkgcache/bus
+install -m 755 -d $RPM_BUILD_ROOT%{pkgcache}/bus
 # `rpm -Vaq ibus` compare st_mode of struct stat with lstat(2) and
 # st_mode of the RPM cache and if the file does not exist, st_mode of
 # RPM cache is o0100000 while the actual st_mode is o0100644.
-touch $RPM_BUILD_ROOT%pkgcache/bus/registry
+touch $RPM_BUILD_ROOT%{pkgcache}/bus/registry
 
 # install .desktop files
 echo "NoDisplay=true" >> $RPM_BUILD_ROOT%{_datadir}/applications/org.freedesktop.IBus.Setup.desktop
@@ -423,12 +423,6 @@ make check \
 
 %postun
 if [ "$1" -eq 0 ]; then
-  # ibus 1.5.31 has no ibus-xinit and need to delete %%_xinputconf here
-  # for the back compatiblity for a year.
-  %{_sbindir}/alternatives --remove xinputrc %{_xinputconf} || :
-  # if alternative was set to manual, reset to auto
-  [ -L %{_sysconfdir}/alternatives/xinputrc -a "`readlink %{_sysconfdir}/alternatives/xinputrc`" = "%{_xinputconf}" ] && %{_sbindir}/alternatives --auto xinputrc || :
-
   # 'dconf update' sometimes does not update the db...
   dconf update || :
   [ -f %{_sysconfdir}/dconf/db/ibus ] && \
@@ -445,17 +439,13 @@ fi
 %posttrans
 dconf update || :
 
-# see https://bugzilla.redhat.com/show_bug.cgi?id=2439813
-# for use of `env -i`
 %transfiletriggerin -- %{_datadir}/ibus/component
 [ -x %{_bindir}/ibus ] && \
-  env -i %{_bindir}/ibus write-cache --system &>/dev/null || :
+  %{_bindir}/ibus write-cache --system &>/dev/null || :
 
-# see https://bugzilla.redhat.com/show_bug.cgi?id=2439813
-# for use of `env -i`
 %transfiletriggerpostun -- %{_datadir}/ibus/component
 [ -x %{_bindir}/ibus ] && \
-  env -i %{_bindir}/ibus write-cache --system &>/dev/null || :
+  %{_bindir}/ibus write-cache --system &>/dev/null || :
 
 
 %ldconfig_scriptlets libs
@@ -478,7 +468,6 @@ dconf update || :
 %{_datadir}/ibus/dicts
 %dir %{_datadir}/ibus/engine
 %{_datadir}/ibus/keymaps
-%{_datadir}/icons/hicolor/*/apps/*
 %{_datadir}/man/man1/ibus.1.gz
 %{_datadir}/man/man1/ibus-daemon.1.gz
 %{_datadir}/man/man7/ibus-emoji.7.gz
@@ -497,10 +486,10 @@ dconf update || :
 %dir %{_prefix}/lib/systemd/user/gnome-session.target.wants
 %{_prefix}/lib/systemd/user/gnome-session.target.wants/*.service
 %{_prefix}/lib/systemd/user/org.freedesktop.IBus.session.*.service
-%verify(not mtime) %dir %pkgcache
-%verify(not mtime) %dir %pkgcache/bus
+%verify(not mtime) %dir %{pkgcache}
+%verify(not mtime) %dir %{pkgcache}/bus
 # 'ibus write-cache --system' updates the system cache.
-%ghost %pkgcache/bus/registry
+%ghost %{pkgcache}/bus/registry
 
 %files -n python3-ibus
 %{python3_sitearch}/gi/overrides/__pycache__/*.py*
@@ -525,12 +514,11 @@ dconf update || :
 %{_libdir}/gtk-4.0/%{gtk4_binary_version}/immodules/libim-ibus.so
 %endif
 
-# The setup package won't include icon files so that
-# gtk-update-icon-cache is executed in the main package only one time.
 %files setup
 %{_bindir}/ibus-setup
 %{_datadir}/applications/org.freedesktop.IBus.Setup.desktop
 %{_datadir}/ibus/setup
+%{_datadir}/icons/hicolor/*/apps/*
 %{_datadir}/man/man1/ibus-setup.1.gz
 
 %files wayland
