diff --git OLD/kbd.spec NEW/kbd.spec
index fd9e681..d95bc31 100644
--- OLD/kbd.spec
+++ NEW/kbd.spec
@@ -40,9 +40,9 @@ Patch8:         kbd-2.9.0-fix-setfont-segfault.patch
 
 BuildRequires:  gcc, bison, flex, gettext, pam-devel, check-devel, automake
 BuildRequires:  console-setup, xkeyboard-config
-BuildRequires: make
+BuildRequires:  make
 Requires:       %{name}-misc = %{version}-%{release}
-Requires:       %{name}-legacy = %{version}-%{release}
+Recommends:     %{name}-legacy = %{version}-%{release}
 # Be sure that system is after UsrMove
 Conflicts:      filesystem < 3
 Provides:       vlock = %{version}
