diff --git OLD/kbd.spec NEW/kbd.spec
index 035057a..ab6bb1a 100644
--- OLD/kbd.spec
+++ NEW/kbd.spec
@@ -37,9 +37,9 @@ Patch7:         kbd-2.0.4-vlock-add-prompt-option.patch
 
 BuildRequires:  gcc, bison, flex, gettext, pam-devel, check-devel, automake
 BuildRequires:  console-setup, xkeyboard-config
-BuildRequires: make
+BuildRequires:  make
 Requires:       %{name}-misc = %{version}-%{release}
-Requires:       %{name}-legacy = %{version}-%{release}
+Recommends:     %{name}-legacy = %{version}-%{release}
 # Be sure that system is after UsrMove
 Conflicts:      filesystem < 3
 Provides:       vlock = %{version}
