diff --git OLD/kmod.spec NEW/kmod.spec
index f91d5b1..90bcc79 100644
--- OLD/kmod.spec
+++ NEW/kmod.spec
@@ -84,6 +84,8 @@ BuildRequires:  make automake libtool
 BuildRequires:  libzstd-devel
 %endif
 
+Requires:	(%{name}-fish-completion = %{version}-%{release} if fish)
+
 Provides:	module-init-tools = 4.0-1
 Obsoletes:	module-init-tools < 4.0-1
 Provides:	/sbin/modprobe
@@ -113,9 +115,19 @@ Summary:	Libraries to handle kernel module loading and unloading
 The kmod-libs package provides runtime libraries for any application that
 wishes to load or unload Linux kernel modules from the running system.
 
+%package fish-completion
+Summary:  Fish shell completion for %{name}
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: fish
+
+BuildArch:  noarch
+
+%description fish-completion
+This package contains fish shell completion for %{name}.
+
 %package devel
 Summary:	Header files for kmod development
-Requires:	%{name}-libs%{?_isa} = %{version}-%{release}
+Requires:	%{name}%{?_isa} = %{version}-%{release}
 
 %description devel
 The kmod-devel package provides header files used for development of
@@ -177,12 +189,10 @@ install -m 0644 %{SOURCE2} $RPM_BUILD_ROOT%{_sysconfdir}/depmod.d/dist.conf
 %{_sbindir}/weak-modules
 %endif
 %{_datadir}/bash-completion/
-%{_datadir}/fish/vendor_functions.d/*
 %{_datadir}/zsh/site-functions/*
 %if %{with dist_conf}
 %{_sysconfdir}/depmod.d/dist.conf
 %endif
-%{_datadir}/pkgconfig/kmod.pc
 %attr(0644,root,root) %{_mandir}/man5/mod*.d*.5*
 %attr(0644,root,root) %{_mandir}/man5/depmod.d.5*
 %{_mandir}/man5/modprobe.conf.5*
@@ -193,8 +203,12 @@ install -m 0644 %{SOURCE2} $RPM_BUILD_ROOT%{_sysconfdir}/depmod.d/dist.conf
 %license COPYING
 %{_libdir}/libkmod.so.*
 
+%files fish-completion
+%{_datadir}/fish/vendor_functions.d/*
+
 %files devel
 %{_includedir}/libkmod.h
+%{_datadir}/pkgconfig/kmod.pc
 %{_libdir}/pkgconfig/libkmod.pc
 %{_libdir}/libkmod.so
 
