--- libical.spec	2025-04-13 23:52:02.705510890 +0800
+++ libical.spec.new	2025-04-13 23:51:01.671915596 +0800
@@ -102,7 +102,6 @@
 %{_libdir}/libicalvcal.so.3
 %{_libdir}/libicalvcal.so.%{version}
 %{_libdir}/girepository-1.0/ICal-3.0.typelib
-%{_datadir}/gir-1.0/ICal-3.0.gir
 
 %files devel
 %doc doc/UsingLibical.md
@@ -114,6 +113,7 @@
 %{_libdir}/pkgconfig/libical.pc
 %{_libdir}/cmake/LibIcal/
 %{_includedir}/libical/
+%{_datadir}/gir-1.0/ICal-3.0.gir
 
 %ldconfig_scriptlets glib
 
@@ -121,12 +121,12 @@
 %{_libdir}/libical-glib.so.3
 %{_libdir}/libical-glib.so.%{version}
 %{_libdir}/girepository-1.0/ICalGLib-3.0.typelib
-%{_datadir}/gir-1.0/ICalGLib-3.0.gir
 
 %files glib-devel
 %{_libdir}/libical-glib.so
 %{_libdir}/pkgconfig/libical-glib.pc
 %{_includedir}/libical-glib/
+%{_datadir}/gir-1.0/ICalGLib-3.0.gir
 %{_datadir}/vala/vapi/libical-glib.vapi
 
 %files glib-doc
