diff --git OLD/libinput.spec NEW/libinput.spec
index 52c7ac1..8dbbd60 100644
--- OLD/libinput.spec
+++ NEW/libinput.spec
@@ -3,6 +3,9 @@
 #global gitdate 20141211
 %global gitversion 58abea394
 
+# Disable libwacom for all builds
+%bcond libwacom 0
+
 Name:           libinput
 Version:        1.31.1
 Release:        1%{?gitdate:.%{gitdate}git%{gitversion}}%{?dist}
@@ -25,12 +28,16 @@ BuildRequires:  meson
 BuildRequires:  pkgconfig(libudev)
 BuildRequires:  pkgconfig(mtdev) >= 1.1.0
 BuildRequires:  pkgconfig(libevdev) >= 0.4
+%if %{with libwacom}
 BuildRequires:  pkgconfig(libwacom) >= 0.20
+%endif
 BuildRequires:  pkgconfig(udev)
 BuildRequires:  python3-rpm-macros
 
 BuildRequires:  lua-devel
 
+Requires:       (%{name}-zsh-completion = %{version}-%{release} if zsh)
+
 %description
 libinput is a library that handles input devices for display servers and other
 applications that need to directly deal with input devices.
@@ -57,6 +64,16 @@ Requires:       python3-pyudev python3-libevdev
 The %{name}-utils package contains tools to debug hardware and analyze
 %{name}.
 
+%package        zsh-completion
+Summary:        Zsh shell completion for %{name}
+Requires:       %{name} = %{version}-%{release}
+Requires:       zsh
+
+BuildArch:      noarch
+
+%description    zsh-completion
+This package contains zsh shell completion for %{name}.
+
 %package        test
 Summary:        libinput integration test suite
 Requires:       %{name}%{?_isa} = %{version}-%{release}
@@ -76,7 +93,13 @@ intended to be run by users.
        -Dtests=true \
        -Dinstall-tests=true \
        -Dautoload-plugins=true \
-       -Dudev-dir=%{udevdir}
+       -Dudev-dir=%{udevdir} \
+%if %{with libwacom}
+       -Dlibwacom=true \
+%else
+       -Dlibwacom=false \
+%endif
+       %{nil}
 %meson_build
 
 %install
@@ -103,9 +126,6 @@ intended to be run by users.
 %{_mandir}/man1/libinput.1*
 %dir %{_datadir}/libinput
 %{_datadir}/libinput/*.quirks
-%dir %{_datadir}/zsh
-%dir %{_datadir}/zsh/site-functions
-%{_datadir}/zsh/site-functions/*
 %{_mandir}/man1/libinput-list-devices.1*
 %{_mandir}/man1/libinput-debug-events.1*
 
@@ -152,6 +172,9 @@ intended to be run by users.
 %{_mandir}/man1/libinput-record.1*
 %{_mandir}/man1/libinput-replay.1*
 
+%files zsh-completion
+%{_datadir}/zsh/site-functions/*
+
 %files test
 %{_libexecdir}/libinput/libinput-test
 %{_libexecdir}/libinput/libinput-test-suite
