diff --git OLD/libnma.spec NEW/libnma.spec
index 1e2ed59..3dd822b 100644
--- OLD/libnma.spec
+++ NEW/libnma.spec
@@ -1,6 +1,3 @@
-%global gtk3_version          %(pkg-config --modversion gtk+-3.0 2>/dev/null || echo bad)
-%global gtk4_version          %(pkg-config --modversion gtk4 2>/dev/null || echo bad)
-%global glib2_version         %(pkg-config --modversion glib-2.0 2>/dev/null || echo bad)
 %global nm_version            1:1.8.0
 %global mbp_version           0.20090602
 %global old_libnma_version    1.10.4
@@ -44,18 +41,27 @@ BuildRequires:  iso-codes-devel
 BuildRequires:  gcr-devel
 BuildRequires:  mobile-broadband-provider-info-devel >= %{mbp_version}
 
+Requires:       %{name}-common = %{version}-%{release}
+
 %description
 This package contains the library used for integrating GUI tools with
 NetworkManager.
 
 
+%package common
+Summary:        Common files for NetworkManager GUI library
+BuildArch:      noarch
+
+%description common
+This package contains common files for the NetworkManager GUI library.
+
+
 %package devel
 Summary:        Header files for NetworkManager GUI library
 Requires:       NetworkManager-libnm-devel >= %{nm_version}
 Obsoletes:      NetworkManager-gtk-devel < 1:0.9.7
 Requires:       libnma%{?_isa} = %{version}-%{release}
 Requires:       gtk3-devel%{?_isa}
-Requires:       pkgconfig
 Conflicts:      libnma < %{old_libnma_version}
 
 %description devel
@@ -63,10 +69,11 @@ This package contains header and pkg-config files to be used for integrating
 GUI tools with NetworkManager.
 
 
+%if %{with libnma_gtk4}
 %package gtk4
 Summary:        Experimental GTK 4 version of NetworkManager GUI library
-Requires:       gtk4%{?_isa} >= %{gtk4_version}
 Requires:       mobile-broadband-provider-info >= %{mbp_version}
+Requires:       %{name}-common = %{version}-%{release}
 Conflicts:      libnma < %{old_libnma_version}
 
 %description gtk4
@@ -79,12 +86,12 @@ Summary:        Header files for experimental GTK4 version of NetworkManager GUI
 Requires:       NetworkManager-libnm-devel >= %{nm_version}
 Requires:       libnma-gtk4%{?_isa} = %{version}-%{release}
 Requires:       gtk4-devel%{?_isa}
-Requires:       pkgconfig
 Conflicts:      libnma < %{old_libnma_version}
 
 %description gtk4-devel
 This package contains the experimental GTK4 version of header and pkg-config
 files to be used for integrating GUI tools with NetworkManager.
+%endif
 
 
 %prep
@@ -112,9 +119,12 @@ files to be used for integrating GUI tools with NetworkManager.
 %meson_test
 
 
-%files -f %{name}.lang
+%files
 %{_libdir}/libnma.so.*
 %{_libdir}/girepository-1.0/NMA-1.0.typelib
+
+
+%files common -f %{name}.lang
 %exclude %{_datadir}/glib-2.0/schemas/org.gnome.nm-applet.gschema.xml
 %{_datadir}/glib-2.0/schemas/org.gnome.nm-applet.eap.gschema.xml
 %doc NEWS CONTRIBUTING
@@ -133,7 +143,6 @@ files to be used for integrating GUI tools with NetworkManager.
 %files gtk4
 %{_libdir}/libnma-gtk4.so.*
 %{_libdir}/girepository-1.0/NMA4-1.0.typelib
-%license COPYING
 
 
 %files gtk4-devel
