From eb1bee192e88acf4616fe351a833580a10bca5c6 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@gnome.org>
Date: Mar 02 2026 14:04:13 +0000
Subject: Fix mingw localizations appearing in main package


---

diff --git a/libsoup3.spec b/libsoup3.spec
index deb949f..10d5bf4 100644
--- a/libsoup3.spec
+++ b/libsoup3.spec
@@ -149,14 +149,14 @@ This is the MinGW build of libsoup3
 %meson_install
 install -m 644 -D tests/libsoup.supp %{buildroot}%{_datadir}/libsoup-3.0/libsoup.supp
 
+%find_lang libsoup-3.0
+
 %if %{with_mingw}
 %mingw_ninja_install
 %mingw_find_lang libsoup-3.0
 %mingw_debug_install_post
 %endif
 
-%find_lang libsoup-3.0
-
 %ifnarch s390x
 %check
 %meson_test

