diff --git OLD/libvncserver.spec NEW/libvncserver.spec
index 083d585..1417933 100644
--- OLD/libvncserver.spec
+++ NEW/libvncserver.spec
@@ -70,9 +70,17 @@ a frame-buffer via the Remote Frame Buffer protocol) easy.
 It hides the programmer from the tedious task of managing clients and
 compression schemata.
 
+%package -n libvncclient
+Summary:    Library to make writing a VNC client easy
+
+%description -n libvncclient
+LibVNCClient makes writing a VNC client (or more correctly, a program importing
+a frame-buffer via the Remote Frame Buffer protocol) easy.
+
 %package devel
 Summary:    Development files for %{name}
 Requires:   %{name}%{?_isa} = %{version}-%{release}
+Requires:   libvncclient%{?_isa} = %{version}-%{release}
 # libvncserver-config deps
 Requires:   coreutils
 # /usr/include/rfb/rfbproto.h:#include <zlib.h>
@@ -122,11 +130,15 @@ done
 %files
 %license COPYING
 %doc AUTHORS ChangeLog NEWS* README* CONTRIBUTING.md HISTORY.md SECURITY.md
-%{_libdir}/libvncclient.so.1
-%{_libdir}/libvncclient.so.%{version}
 %{_libdir}/libvncserver.so.1
 %{_libdir}/libvncserver.so.%{version}
 
+%files -n libvncclient
+%license COPYING
+%doc AUTHORS ChangeLog NEWS* README* CONTRIBUTING.md HISTORY.md SECURITY.md
+%{_libdir}/libvncclient.so.1
+%{_libdir}/libvncclient.so.%{version}
+
 %files devel
 %{_includedir}/rfb/
 %{_libdir}/libvncclient.so
