diff --git OLD/localsearch.spec NEW/localsearch.spec
index a8193d2..7e219fa 100644
--- OLD/localsearch.spec
+++ NEW/localsearch.spec
@@ -11,9 +11,11 @@
 %endif
 
 %global with_enca 1
-%global with_ffmpeg 1
+%global with_ffmpeg 0
+%global with_iso 0
 %global with_libcue 1
-%global with_totem_pl_parser 1
+%global with_totem_pl_parser 0
+%global with_xps 0
 
 %if 0%{?rhel}
 %global with_enca 0
@@ -70,7 +72,9 @@ BuildRequires:  pkgconfig(libcue)
 %endif
 BuildRequires:  pkgconfig(libexif)
 BuildRequires:  pkgconfig(libgsf-1)
+%if 0%{?with_xps}
 BuildRequires:  pkgconfig(libgxps)
+%endif
 BuildRequires:  pkgconfig(libjpeg)
 BuildRequires:  pkgconfig(libpng)
 BuildRequires:  pkgconfig(libseccomp)
@@ -84,7 +88,9 @@ BuildRequires:  pkgconfig(totem-plparser)
 BuildRequires:  pkgconfig(tinysparql-3.0) >= %{tinysparql_version}
 BuildRequires:  pkgconfig(vorbisfile)
 %if !0%{?flatpak}
+%if 0%{?with_iso}
 BuildRequires:  pkgconfig(libosinfo-1.0)
+%endif
 BuildRequires:  pkgconfig(libnm)
 BuildRequires:  pkgconfig(upower-glib)
 %endif
@@ -102,6 +108,14 @@ tag/metadata database and search tool.
 This package contains various miners and metadata extractors for tinysparql.
 
 
+%package test
+Summary:        Localsearch test utilities
+Requires:       %{name} = %{version}-%{release}
+
+%description test
+This package contains test utilities for the Localsearch miners.
+
+
 %prep
 %autosetup -p1 -n %{name}-%{tarball_version}
 
@@ -126,6 +140,12 @@ This package contains various miners and metadata extractors for tinysparql.
 %endif
 %if ! 0%{?with_totem_pl_parser}
   -Dplaylist=disabled \
+%endif
+%if ! 0%{?with_xps}
+  -Dxps=disabled \
+%endif
+%if ! 0%{?with_iso}
+  -Diso=disabled \
 %endif
   %{nil}
 
@@ -154,6 +174,7 @@ This package contains various miners and metadata extractors for tinysparql.
 %config(noreplace) %{_sysconfdir}/xdg/autostart/localsearch-3.desktop
 %{_bindir}/localsearch
 %{_libdir}/localsearch-3.0/
+%exclude %{_libdir}/localsearch-3.0/trackertestutils/
 %{_libexecdir}/localsearch-3
 %{_libexecdir}/localsearch-control-3
 %{_libexecdir}/localsearch-extractor-3
@@ -182,6 +203,9 @@ This package contains various miners and metadata extractors for tinysparql.
 %{_userunitdir}/localsearch*.service
 %endif
 
+%files test
+%{_libdir}/localsearch-3.0/trackertestutils/
+
 
 %changelog
 %autochangelog
