diff --git OLD/localsearch.spec NEW/localsearch.spec
index 2108175..8e9df7d 100644
--- OLD/localsearch.spec
+++ NEW/localsearch.spec
@@ -8,9 +8,11 @@
 %endif
 
 %global with_enca 1
-%global with_ffmpeg 1
+%global with_ffmpeg 0
+%global with_iso 0
 %global with_libcue 1
-%global with_totem_pl_parser 1
+%global with_totem_pl_parser 0
+%global with_xps 0
 
 %if 0%{?rhel} || 0%{?flatpak}
 %global with_enca 0
@@ -68,7 +70,9 @@ BuildRequires:  pkgconfig(libcue)
 %endif
 BuildRequires:  pkgconfig(libexif)
 BuildRequires:  pkgconfig(libgsf-1)
+%if 0%{?with_xps}
 BuildRequires:  pkgconfig(libgxps)
+%endif
 BuildRequires:  pkgconfig(libiptcdata)
 BuildRequires:  pkgconfig(libjpeg)
 BuildRequires:  pkgconfig(libpng)
@@ -82,7 +86,9 @@ BuildRequires:  pkgconfig(totem-plparser)
 BuildRequires:  pkgconfig(tinysparql-3.0) >= %{tinysparql_version}
 BuildRequires:  pkgconfig(vorbisfile)
 %if !0%{?flatpak}
+%if 0%{?with_iso}
 BuildRequires:  pkgconfig(libosinfo-1.0)
+%endif
 BuildRequires:  pkgconfig(libnm)
 BuildRequires:  pkgconfig(upower-glib)
 %endif
@@ -100,6 +106,14 @@ tag/metadata database and search tool.
 This package contains various miners and metadata extractors for tinysparql.
 
 
+%package test
+Summary:        Localsearch test utilities
+Requires:       %{name} = %{version}-%{release}
+
+%description test
+This package contains test utilities for the Localsearch miners.
+
+
 %prep
 %autosetup -p1 -n %{name}-%{tarball_version}
 
@@ -124,6 +138,12 @@ This package contains various miners and metadata extractors for tinysparql.
 %endif
 %if ! 0%{?with_totem_pl_parser}
   -Dplaylist=disabled \
+%endif
+%if ! 0%{?with_xps}
+  -Dxps=disabled \
+%endif
+%if ! 0%{?with_iso}
+  -Diso=disabled \
 %endif
   %{nil}
 
@@ -161,6 +181,7 @@ ln -sr %{buildroot}%{_datadir}/dbus-1/services/%{domain_ontology}.Tracker3.Miner
 %config(noreplace) %{_sysconfdir}/xdg/autostart/localsearch-3.desktop
 %{_bindir}/localsearch
 %{_libdir}/localsearch-3.0/
+%exclude %{_libdir}/localsearch-3.0/trackertestutils/
 %{_libexecdir}/localsearch-3
 %{_libexecdir}/localsearch-control-3
 %{_libexecdir}/localsearch-extractor-3
@@ -192,6 +213,9 @@ ln -sr %{buildroot}%{_datadir}/dbus-1/services/%{domain_ontology}.Tracker3.Miner
 %{_bindir}/%{name}-flatpak-fixup.sh
 %endif
 
+%files test
+%{_libdir}/localsearch-3.0/trackertestutils/
+
 
 %changelog
 %autochangelog
