diff --git OLD/lvm2.spec NEW/lvm2.spec
index 3a7b69f..e0308d1 100644
--- OLD/lvm2.spec
+++ NEW/lvm2.spec
@@ -89,7 +89,7 @@ Requires: %{name}-libs = %{?epoch}:%{version}-%{release}
 Requires(post): (system-release >= %{system_release_version} if system-release)
 %endif
 Requires: bash >= %{bash_version}
-Requires(post): systemd-units >= %{systemd_version}, systemd-sysv
+Requires(post): systemd-units >= %{systemd_version}
 Requires(preun): systemd-units >= %{systemd_version}
 Requires(postun): systemd-units >= %{systemd_version}
 Requires: module-init-tools
@@ -222,12 +222,6 @@ systemctl start lvm2-lvmpolld.socket >/dev/null 2>&1 || :
 %systemd_postun_with_restart lvm2-lvmpolld.service
 %endif
 
-%triggerun -- %{name} < 2.02.86-2
-%{_bindir}/systemd-sysv-convert --save lvm2-monitor >/dev/null 2>&1 || :
-/bin/systemctl --no-reload enable lvm2-monitor.service > /dev/null 2>&1 || :
-/sbin/chkconfig --del lvm2-monitor > /dev/null 2>&1 || :
-/bin/systemctl try-restart lvm2-monitor.service > /dev/null 2>&1 || :
-
 %files
 %license COPYING COPYING.LIB
 %doc README VERSION WHATS_NEW
