diff --git OLD/ModemManager.spec NEW/ModemManager.spec
index 5992374..036efa2 100644
--- OLD/ModemManager.spec
+++ NEW/ModemManager.spec
@@ -108,6 +108,7 @@ Vala bindings for ModemManager
 	-Ddist_version='"%{version}-%{release}"' \
 	-Dudevdir=/usr/lib/udev \
 	-Dsystemdsystemunitdir=%{_unitdir} \
+	-Ddbus_policy_dir=%{_datadir}/dbus-1/system.d \
 	-Dvapi=true \
 	-Dgtk_doc=true \
 	-Dpolkit=permissive \
@@ -145,7 +146,7 @@ cp -a cli/mmcli-completion %{buildroot}%{_datadir}/bash-completion/completions/m
 %files -f %{name}.lang
 %license COPYING
 %doc README
-%{_sysconfdir}/dbus-1/system.d/org.freedesktop.ModemManager1.conf
+%{_datadir}/dbus-1/system.d/org.freedesktop.ModemManager1.conf
 %{_datadir}/dbus-1/system-services/org.freedesktop.ModemManager1.service
 %attr(0755,root,root) %{_sbindir}/ModemManager
 %attr(0755,root,root) %{_bindir}/mmcli
