diff --git OLD/mutter.spec NEW/mutter.spec
index 0e9cac6..05cf9c3 100644
--- OLD/mutter.spec
+++ NEW/mutter.spec
@@ -46,7 +46,6 @@ BuildRequires: pkgconfig(libsystemd)
 BuildRequires: pkgconfig(umockdev-1.0)
 BuildRequires: desktop-file-utils
 BuildRequires: cvt
-BuildRequires: python3-argcomplete
 BuildRequires: python3-docutils
 # Bootstrap requirements
 BuildRequires: gettext-devel git-core
@@ -69,7 +68,6 @@ BuildRequires: pkgconfig(libeis-1.0) >= %{libei_version}
 
 BuildRequires: pkgconfig(libinput) >= %{libinput_version}
 BuildRequires: pkgconfig(xwayland)
-BuildRequires: pkgconfig(bash-completion)
 
 BuildRequires: python3-dbusmock
 
@@ -81,7 +79,7 @@ Requires: libinput%{?_isa} >= %{libinput_version}
 Requires: pipewire%{_isa} >= %{pipewire_version}
 Requires: startup-notification
 Requires: dbus
-Requires: python3-argcomplete
+Requires: xorg-x11-server-Xwayland%{?_isa} >= 1.20.99.1
 
 # Need common
 Requires: %{name}-common = %{version}-%{release}
@@ -145,7 +143,9 @@ the functionality of the installed %{name} package.
 %autosetup -S git -n %{name}-%{tarball_version}
 
 %build
-%meson -Degl_device=true
+%meson -Degl_device=true \
+        -Dbash_completion=false \
+        %{nil}
 %meson_build
 
 %install
@@ -161,9 +161,9 @@ install -p %{SOURCE1} %{buildroot}%{_datadir}/glib-2.0/schemas
 %{_datadir}/polkit-1/actions/org.gnome.mutter.*.policy
 %{_bindir}/gdctl
 %{_bindir}/gnome-service-client
-%{_datadir}/bash-completion/completions/gdctl
 %{_libdir}/lib*.so.*
 %{_libdir}/mutter-%{mutter_api_version}/
+%exclude %{_libdir}/mutter-%{mutter_api_version}/*.gir
 %{_libexecdir}/mutter-backlight-helper
 %{_libexecdir}/mutter-x11-frames
 %{_mandir}/man1/mutter.1*
