diff --git OLD/mutter.spec NEW/mutter.spec
index cf53a39..cde91b1 100644
--- OLD/mutter.spec
+++ NEW/mutter.spec
@@ -17,6 +17,8 @@
 %global major_version %%(echo %{version} | cut -d '.' -f1 | cut -d '~' -f 1)
 %global tarball_version %%(echo %{version} | tr '~' '.')
 
+%bcond_with completion
+
 Name:          mutter
 Version:       49.2
 Release:       %autorelease
@@ -41,7 +43,9 @@ BuildRequires: mesa-libGLES-devel
 BuildRequires: mesa-libGL-devel
 BuildRequires: mesa-libgbm-devel
 BuildRequires: pam-devel
+%if %{with completion}
 BuildRequires: pkgconfig(bash-completion)
+%endif
 BuildRequires: pkgconfig(colord) >= %{colord_version}
 BuildRequires: pkgconfig(glib-2.0) >= %{glib_version}
 BuildRequires: pkgconfig(gobject-introspection-1.0) >= %{gobject_introspection_version}
@@ -57,7 +61,9 @@ BuildRequires: pkgconfig(libpipewire-0.3) >= %{pipewire_version}
 BuildRequires: pkgconfig(sysprof-capture-4)
 BuildRequires: pkgconfig(libsystemd)
 BuildRequires: pkgconfig(umockdev-1.0)
+%if %{with completion}
 BuildRequires: python3-argcomplete
+%endif
 BuildRequires: python3-docutils
 # Bootstrap requirements
 BuildRequires: gettext-devel git-core
@@ -94,7 +100,10 @@ Requires: libinput%{?_isa} >= %{libinput_version}
 Requires: pipewire%{_isa} >= %{pipewire_version}
 Requires: startup-notification
 Requires: dbus
+%if %{with completion}
 Requires: python3-argcomplete
+%endif
+Requires: xorg-x11-server-Xwayland%{?_isa} >= 1.20.99.1
 
 # Need common
 Requires: %{name}-common = %{version}-%{release}
@@ -160,7 +169,13 @@ the functionality of the installed %{name} package.
 %autosetup -S git -n %{name}-%{tarball_version}
 
 %build
-%meson -Degl_device=true
+%meson -Degl_device=true \
+%if %{with completion}
+        -Dbash_completion=true \
+%else
+        -Dbash_completion=false \
+%endif
+        %{nil}
 %meson_build
 
 %install
@@ -184,7 +199,9 @@ install -p %{SOURCE1} %{buildroot}%{_datadir}/glib-2.0/schemas
 %{_mandir}/man1/mutter.1*
 %{_mandir}/man1/gdctl.1*
 %{_mandir}/man1/gnome-service-client.1*
+%if %{with completion}
 %{bash_completions_dir}/gdctl
+%endif
 
 %files common
 %{_datadir}/GConf/gsettings/mutter-schemas.convert
