diff --git OLD/mutter.spec NEW/mutter.spec
index b5874fb..c56f58b 100644
--- OLD/mutter.spec
+++ NEW/mutter.spec
@@ -63,7 +63,6 @@ BuildRequires: pkgconfig(libsystemd)
 BuildRequires: pkgconfig(xkeyboard-config)
 BuildRequires: desktop-file-utils
 BuildRequires: cvt
-BuildRequires: python3-argcomplete
 BuildRequires: python3-docutils
 # Bootstrap requirements
 BuildRequires: gettext-devel git-core
@@ -96,7 +95,6 @@ Requires: libinput%{?_isa} >= %{libinput_version}
 Requires: pipewire%{_isa} >= %{pipewire_version}
 Requires: startup-notification
 Requires: dbus
-Requires: python3-argcomplete
 
 # Need common
 Requires: %{name}-common = %{version}-%{release}
@@ -160,7 +158,10 @@ the functionality of the installed %{name} package.
 %autosetup -S git -n %{name}-%{tarball_version}
 
 %build
-%meson -Degl_device=true
+%meson -Degl_device=true \
+        -Dbash_completion=false \
+        -Dx11=false \
+        %{nil}
 %meson_build
 
 %install
@@ -175,12 +176,11 @@ install -p %{SOURCE1} %{buildroot}%{_datadir}/glib-2.0/schemas
 %{_bindir}/mutter
 %{_libdir}/lib*.so.*
 %{_libdir}/mutter-%{mutter_api_version}/
-%{_libexecdir}/mutter-restart-helper
-%{_libexecdir}/mutter-x11-frames
+%exclude %{_libdir}/mutter-%{mutter_api_version}/*.gir
+%{_libexecdir}/mutter-*
 %{_mandir}/man1/mutter.1*
 %{_bindir}/gdctl
 %{_mandir}/man1/gdctl.1*
-%{_sysconfdir}/bash_completion.d/gdctl
 
 %files common
 %{_datadir}/GConf/gsettings/mutter-schemas.convert
@@ -193,6 +193,7 @@ install -p %{SOURCE1} %{buildroot}%{_datadir}/glib-2.0/schemas
 %files devel
 %{_includedir}/*
 %{_libdir}/lib*.so
+%{_libdir}/mutter-%{mutter_api_version}/*.gir
 %{_libdir}/pkgconfig/*
 
 %files tests
