diff --git OLD/mutter.spec NEW/mutter.spec
index 302d160..fbd5f5e 100644
--- OLD/mutter.spec
+++ NEW/mutter.spec
@@ -27,6 +27,7 @@ License:       GPL-2.0-or-later
 URL:           http://www.gnome.org
 Source0:       http://download.gnome.org/sources/%{name}/%{major_version}/%{name}-%{tarball_version}.tar.xz
 
+Patch:         mutter-writable-system-config-native-gdctl.patch
 
 BuildRequires: cvt
 BuildRequires: desktop-file-utils
@@ -51,7 +52,6 @@ BuildRequires: pkgconfig(libpipewire-0.3) >= %{pipewire_version}
 BuildRequires: pkgconfig(sysprof-capture-4)
 BuildRequires: pkgconfig(libsystemd)
 BuildRequires: pkgconfig(umockdev-1.0)
-BuildRequires: python3-argcomplete
 BuildRequires: python3-docutils
 # Bootstrap requirements
 BuildRequires: gettext-devel git-core
@@ -88,7 +88,7 @@ Requires: libinput%{?_isa} >= %{libinput_version}
 Requires: pipewire%{_isa} >= %{pipewire_version}
 Requires: startup-notification
 Requires: dbus
-Requires: python3-argcomplete
+Requires: xorg-x11-server-Xwayland%{?_isa} >= 1.20.99.1
 
 # Need common
 Requires: %{name}-common = %{version}-%{release}
@@ -173,7 +173,6 @@ Viewer for nested mutter instances.
 %license COPYING
 %doc NEWS
 %{_bindir}/gdctl
-%{_bindir}/gnome-service-client
 %{_bindir}/mutter
 %{_datadir}/polkit-1/actions/org.gnome.mutter.*.policy
 %{_libdir}/lib*.so.*
@@ -183,7 +182,6 @@ Viewer for nested mutter instances.
 %{_libexecdir}/mutter-x11-frames
 %{_mandir}/man1/mutter.1*
 %{_mandir}/man1/gdctl.1*
-%{_mandir}/man1/gnome-service-client.1*
 %{bash_completions_dir}/gdctl
 
 %files common
@@ -205,10 +203,12 @@ Viewer for nested mutter instances.
 %{_libexecdir}/installed-tests/mutter-%{mutter_api_version}
 
 %files devkit
+%{_bindir}/gnome-service-client
 %{_datadir}/applications/org.gnome.Mutter.Mdk.desktop
 %{_datadir}/glib-2.0/schemas/org.gnome.mutter.devkit.gschema.xml
 %{_datadir}/icons/hicolor/*/apps/org.gnome.Mutter.Mdk*
 %{_libexecdir}/mutter-devkit
+%{_mandir}/man1/gnome-service-client.1*
 
 %changelog
 %autochangelog
