--- nautilus.spec	2025-01-23 20:29:10.098261359 +0800
+++ nautilus.spec.new	2025-01-23 21:48:18.538594710 +0800
@@ -18,8 +18,6 @@
 License:        GPL-3.0-or-later AND CC0-1.0
 URL:            https://apps.gnome.org/Nautilus/
 Source0:        https://download.gnome.org/sources/%{name}/%{major_version}/%{name}-%{tarball_version}.tar.xz
-# https://pagure.io/fedora-workstation/issue/442
-Patch:          default-terminal.patch
 
 BuildRequires:  desktop-file-utils
 BuildRequires:  gcc
@@ -130,8 +128,6 @@
 %{_datadir}/glib-2.0/schemas/org.gnome.nautilus.gschema.xml
 %{_datadir}/nautilus/
 %exclude %{_datadir}/localsearch3/domain-ontologies/org.gnome.Nautilus.domain.rule
-%{_libdir}/nautilus/extensions-4/libnautilus-image-properties.so
-%{_libdir}/nautilus/extensions-4/libtotem-properties-page.so
 %{_metainfodir}/org.gnome.Nautilus.metainfo.xml
 
 %files extensions
@@ -140,6 +136,8 @@
 %{_libdir}/girepository-1.0/Nautilus-4.0.typelib
 %dir %{_libdir}/nautilus
 %dir %{_libdir}/nautilus/extensions-4
+%{_libdir}/nautilus/extensions-4/libnautilus-image-properties.so
+%{_libdir}/nautilus/extensions-4/libtotem-properties-page.so
 
 %files devel
 %{_includedir}/nautilus
