diff --git OLD/NetworkManager.spec NEW/NetworkManager.spec
index 2bb459e..bcee091 100644
--- OLD/NetworkManager.spec
+++ NEW/NetworkManager.spec
@@ -185,6 +185,7 @@ Requires(postun): systemd
 Requires: dbus >= %{dbus_version}
 Requires: glib2 >= %{glib2_version}
 Requires: %{name}-libnm%{?_isa} = %{epoch}:%{version}-%{release}
+Requires: (%{name}-firewalld = %{epoch}:%{version}-%{release} if firewalld)
 
 Recommends: iputils
 
@@ -314,6 +315,17 @@ Ethernet, Bridge, Bond, VLAN, Team, InfiniBand, Wi-Fi, mobile broadband
 services.
 
 
+%package firewalld
+Summary: NetworkManager firewalld rules
+Requires: %{name} = %{epoch}:%{version}-%{release}
+Requires: firewalld
+
+BuildArch: noarch
+
+%description firewalld
+This package contains NetworkManager firewalld rules.
+
+
 %if %{with adsl}
 %package adsl
 Summary: ADSL device plugin for NetworkManager
@@ -789,11 +801,14 @@ if [ -S /run/udev/control ]; then
     /usr/bin/udevadm control --reload-rules || :
     /usr/bin/udevadm trigger --subsystem-match=net || :
 fi
-%firewalld_reload
 
 %systemd_post %{systemd_units}
 
 
+%post firewalld
+%firewalld_reload
+
+
 %if %{with ifcfg_rh}
 %post initscripts-updown
 if [ -f %{_sbindir}/ifup -a ! -L %{_sbindir}/ifup ]; then
@@ -840,11 +855,14 @@ fi
 %postun
 /usr/bin/udevadm control --reload-rules || :
 /usr/bin/udevadm trigger --subsystem-match=net || :
-%firewalld_reload
 
 %systemd_postun %{systemd_units}
 
 
+%postun firewalld
+%firewalld_reload
+
+
 %if %{with nm_cloud_setup}
 %postun cloud-setup
 %systemd_postun %{systemd_units_cloud_setup}
@@ -916,7 +934,6 @@ fi
 %{_datadir}/polkit-1/rules.d/org.freedesktop.NetworkManager.rules
 %endif
 %{_prefix}/lib/udev/rules.d/*.rules
-%{_prefix}/lib/firewalld/zones/nm-shared.xml
 # systemd stuff
 %{_unitdir}/NetworkManager.service
 %{_unitdir}/NetworkManager-wait-online.service
@@ -933,6 +950,10 @@ fi
 %license COPYING.GFDL
 
 
+%files firewalld
+%{_prefix}/lib/firewalld/zones/nm-shared.xml
+
+
 %if %{with adsl}
 %files adsl
 %{nmplugindir}/libnm-device-plugin-adsl.so
