diff --git OLD/PackageKit.spec NEW/PackageKit.spec
index b9e1caf..c6620f9 100644
--- OLD/PackageKit.spec
+++ NEW/PackageKit.spec
@@ -178,6 +178,20 @@ Requires: %{name}%{?_isa} = %{version}-%{release}
 %description cron
 Crontab and utilities for running PackageKit as a cron job.
 
+%package utils
+Summary: Command line utilities for PackageKit
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description utils
+Command line utilities for PackageKit.
+
+%package backend-test
+Summary: Backend test utilities for PackageKit
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description backend-test
+Utilities for testing the PackageKit backend.
+
 %package devel
 Summary: Libraries and headers for PackageKit
 Requires: %{name}-glib-devel%{?_isa} = %{version}-%{release}
@@ -296,27 +310,17 @@ systemctl disable packagekit-offline-update.service > /dev/null 2>&1 || :
 %dir %{_localstatedir}/cache/app-info/xmls
 %endif
 %dir %{_localstatedir}/cache/PackageKit
-%{_datadir}/bash-completion/completions/pkcon
-%{_datadir}/bash-completion/completions/pkgcli
 %dir %{_libdir}/packagekit-backend
 %config(noreplace) %{_sysconfdir}/PackageKit/PackageKit.conf
 %config(noreplace) %{_sysconfdir}/PackageKit/Vendor.conf
-%{_datadir}/man/man1/pkgcli.1*
-%{_datadir}/man/man1/pkcon.1*
-%{_datadir}/man/man1/pkmon.1*
 %{_datadir}/polkit-1/actions/*.policy
 %{_datadir}/polkit-1/rules.d/*
 %{_datadir}/PackageKit/pk-upgrade-distro.sh
-%{_datadir}/PackageKit/helpers/test_spawn/search-name.sh
 %{_metainfodir}/org.freedesktop.packagekit.metainfo.xml
 %{_libexecdir}/packagekitd
 %{_libexecdir}/packagekit-direct
-%{_bindir}/pkgcli
-%{_bindir}/pkmon
-%{_bindir}/pkcon
 %exclude %{_libdir}/libpackagekit*.so.*
 %{_libdir}/packagekit-backend/libpk_backend_dummy.so
-%{_libdir}/packagekit-backend/libpk_backend_test_*.so
 %ghost %verify(not md5 size mtime) %attr(0644,-,-) %{_localstatedir}/lib/PackageKit/transactions.db
 %{_datadir}/dbus-1/system.d/*
 %{_datadir}/dbus-1/system-services/*.service
@@ -359,6 +363,22 @@ systemctl disable packagekit-offline-update.service > /dev/null 2>&1 || :
 %config %{_sysconfdir}/cron.daily/packagekit-background.cron
 %config(noreplace) %{_sysconfdir}/sysconfig/packagekit-background
 
+%files utils
+%{_bindir}/pkgcli
+%{_bindir}/pkmon
+%{_bindir}/pkcon
+%{_datadir}/man/man1/pkgcli.1*
+%{_datadir}/man/man1/pkcon.1*
+%{_datadir}/man/man1/pkmon.1*
+%{_datadir}/bash-completion/completions/pkcon
+%{_datadir}/bash-completion/completions/pkgcli
+
+%files backend-test
+%dir %{_datadir}/PackageKit/helpers
+%dir %{_datadir}/PackageKit/helpers/test_spawn
+%{_datadir}/PackageKit/helpers/test_spawn/search-name.sh
+%{_libdir}/packagekit-backend/libpk_backend_test_*.so
+
 %files gstreamer-plugin
 %{_libexecdir}/pk-gstreamer-install
 %{_libexecdir}/gst-install-plugins-helper
