diff --git OLD/pkgconf.spec NEW/pkgconf.spec
index 8d3dbd1..59e97a4 100644
--- OLD/pkgconf.spec
+++ NEW/pkgconf.spec
@@ -105,7 +105,7 @@ Provides:       pkgconfig(pkg-config) = %{version}
 Provides:       pkg-config = %{pkgconfig_verrel}
 Provides:       pkg-config%{?_isa} = %{pkgconfig_verrel}
 Requires:       %{name}%{?_isa} = %{version}-%{release}
-Requires:       %{name}-m4 = %{version}-%{release}
+Requires:       (%{name}-m4 = %{version}-%{release} if m4)
 
 %description pkg-config
 This package provides the shim links for pkgconf to be automatically
