diff --git OLD/policycoreutils.spec NEW/policycoreutils.spec
index e97d3c2..6665853 100644
--- OLD/policycoreutils.spec
+++ NEW/policycoreutils.spec
@@ -63,6 +63,7 @@ Patch0020: 0020-sandbox-seunshare-rewrite-to-pin-directories-before-.patch
 Patch0021: 0021-sandbox-seunshare-fully-check-setfsuid-calls.patch
 Patch0022: 0022-sandbox-seunshare-check-owner-in-seunshare_mount_fil.patch
 Patch0023: 0023-sandbox-seunshare-fix-fd_tmpdir_r-check.patch
+Patch9001: selinux-policycoreutils-fixfiles.patch
 # Patch list end
 
 # gen_changelog
@@ -94,8 +95,8 @@ BuildRequires: (python3-wheel if python3-setuptools < 71)
 BuildRequires: systemd
 BuildRequires: git-core
 BuildRequires: gnupg2
-Requires: util-linux grep gawk diffutils sed
-Requires: libsepol >= %{libsepolver} coreutils libselinux-utils >=  %{libselinuxver}
+# Requires: util-linux grep gawk diffutils sed coreutils
+Requires: libsepol >= %{libsepolver} libselinux-utils >=  %{libselinuxver}
 Recommends: rpm
 
 %description
@@ -298,6 +299,7 @@ Requires: python3-libdnf5
 %else
 Requires: python3-dnf
 %endif
+Requires: policycoreutils-semodule-utils = %{version}-%{release}
 Requires: (selinux-policy-devel if selinux-policy)
 
 %description devel
@@ -322,6 +324,25 @@ The policycoreutils-devel package contains the management tools use to develop p
 %{_usr}/share/bash-completion/completions/sepolicy
 
 
+%package semodule-utils
+Summary: SELinux semodule utilities
+Requires: policycoreutils = %{version}-%{release}
+
+%description semodule-utils
+The policycoreutils-semoudle-utils package contains the semodule utilities
+to manage SELinux policy modules.
+
+%files semodule-utils
+%{_bindir}/semodule_expand
+%{_bindir}/semodule_link
+%{_bindir}/semodule_package
+%{_bindir}/semodule_unpackage
+%{_mandir}/man8/semodule_expand.8*
+%{_mandir}/man8/semodule_link.8*
+%{_mandir}/man8/semodule_unpackage.8*
+%{_mandir}/man8/semodule_package.8*
+
+
 %package sandbox
 Summary: SELinux sandbox utilities
 Requires: python3-policycoreutils = %{version}-%{release}
@@ -410,10 +431,6 @@ system-config-selinux is a utility for managing the SELinux environment
 %{_sbindir}/sestatus
 %endif
 %{_bindir}/secon
-%{_bindir}/semodule_expand
-%{_bindir}/semodule_link
-%{_bindir}/semodule_package
-%{_bindir}/semodule_unpackage
 %{_bindir}/sestatus
 %{_libexecdir}/selinux/hll
 %{_libexecdir}/selinux/selinux-autorelabel
@@ -435,10 +452,6 @@ system-config-selinux is a utility for managing the SELinux environment
 %{_mandir}/man1/secon.1*
 %{_mandir}/man1/unsetfiles.1*
 %{_mandir}/man8/genhomedircon.8*
-%{_mandir}/man8/semodule_expand.8*
-%{_mandir}/man8/semodule_link.8*
-%{_mandir}/man8/semodule_unpackage.8*
-%{_mandir}/man8/semodule_package.8*
 %dir %{_datadir}/bash-completion
 %{_datadir}/bash-completion/completions/setsebool
 %{!?_licensedir:%global license %%doc}
