diff --git OLD/policycoreutils.spec NEW/policycoreutils.spec
index 9fc6c5e..f81c8f2 100644
--- OLD/policycoreutils.spec
+++ NEW/policycoreutils.spec
@@ -46,6 +46,7 @@ Patch0003: 0003-sandbox-Use-matchbox-window-manager-instead-of-openb.patch
 Patch0004: 0004-Use-SHA-2-instead-of-SHA-1.patch
 Patch0005: 0005-python-sepolicy-Fix-spec-file-dependencies.patch
 Patch0006: 0006-sepolicy-Fix-detection-of-writeable-locations.patch
+Patch9001: selinux-policycoreutils-fixfiles.patch
 # Patch list end
 
 # gen_changelog
@@ -77,8 +78,8 @@ BuildRequires: (python3-wheel if python3-setuptools < 71)
 BuildRequires: systemd
 BuildRequires: git-core
 BuildRequires: gnupg2
-Requires: util-linux grep gawk diffutils sed
-Requires: libsepol >= %{libsepolver} coreutils libselinux-utils >=  %{libselinuxver}
+# Requires: util-linux grep gawk diffutils sed coreutils
+Requires: libsepol >= %{libsepolver} libselinux-utils >=  %{libselinuxver}
 Recommends: rpm
 
 %description
@@ -281,6 +282,7 @@ Requires: python3-libdnf5
 %else
 Requires: python3-dnf
 %endif
+Requires: policycoreutils-semodule-utils = %{version}-%{release}
 Requires: (selinux-policy-devel if selinux-policy)
 
 %description devel
@@ -305,6 +307,25 @@ The policycoreutils-devel package contains the management tools use to develop p
 %{_usr}/share/bash-completion/completions/sepolicy
 
 
+%package semodule-utils
+Summary: SELinux semodule utilities
+Requires: policycoreutils = %{version}-%{release}
+
+%description semodule-utils
+The policycoreutils-semoudle-utils package contains the semodule utilities
+to manage SELinux policy modules.
+
+%files semodule-utils
+%{_bindir}/semodule_expand
+%{_bindir}/semodule_link
+%{_bindir}/semodule_package
+%{_bindir}/semodule_unpackage
+%{_mandir}/man8/semodule_expand.8*
+%{_mandir}/man8/semodule_link.8*
+%{_mandir}/man8/semodule_unpackage.8*
+%{_mandir}/man8/semodule_package.8*
+
+
 %package sandbox
 Summary: SELinux sandbox utilities
 Requires: python3-policycoreutils = %{version}-%{release}
@@ -393,10 +414,6 @@ system-config-selinux is a utility for managing the SELinux environment
 %{_sbindir}/sestatus
 %endif
 %{_bindir}/secon
-%{_bindir}/semodule_expand
-%{_bindir}/semodule_link
-%{_bindir}/semodule_package
-%{_bindir}/semodule_unpackage
 %{_bindir}/sestatus
 %{_libexecdir}/selinux/hll
 %{_libexecdir}/selinux/selinux-autorelabel
@@ -418,10 +435,6 @@ system-config-selinux is a utility for managing the SELinux environment
 %{_mandir}/man1/secon.1*
 %{_mandir}/man1/unsetfiles.1*
 %{_mandir}/man8/genhomedircon.8*
-%{_mandir}/man8/semodule_expand.8*
-%{_mandir}/man8/semodule_link.8*
-%{_mandir}/man8/semodule_unpackage.8*
-%{_mandir}/man8/semodule_package.8*
 %dir %{_datadir}/bash-completion
 %{_datadir}/bash-completion/completions/setsebool
 %{!?_licensedir:%global license %%doc}
