diff --git OLD/selinux-policy.spec NEW/selinux-policy.spec
index 9344da2..d339d46 100644
--- OLD/selinux-policy.spec
+++ NEW/selinux-policy.spec
@@ -59,6 +59,7 @@ BuildRequires: groff
 Requires(pre): policycoreutils >= %{POLICYCOREUTILSVER}
 Requires(pre): /usr/bin/sha512sum
 Requires(meta): (rpm-plugin-selinux if rpm-libs)
+Requires(meta): (%{name}-rpm-macros = %{version}-%{release} if rpm-build)
 Requires: selinux-policy-any = %{version}-%{release}
 Provides: selinux-policy-base = %{version}-%{release}
 Provides: selinux-policy-stable = %{STABLEVER}
@@ -73,18 +74,12 @@ the policy has been adjusted to provide support for Fedora.
 %{!?_licensedir:%global license %%doc}
 %license COPYING
 %dir %{_datadir}/selinux
-%dir %{_datadir}/selinux/devel
-%dir %{_datadir}/selinux/devel/include
-%dir %{_datadir}/selinux/devel/include/distributed
 %dir %{_datadir}/selinux/packages
-%dir %{_datadir}/selinux/packages/minimum
-%dir %{_datadir}/selinux/packages/mls
 %dir %{_datadir}/selinux/packages/targeted
 %dir %{_sysconfdir}/selinux
 %ghost %config(noreplace) %{_sysconfdir}/selinux/config
 %ghost %{_sysconfdir}/sysconfig/selinux
 %{_usr}/lib/tmpfiles.d/selinux-policy.conf
-%{_rpmconfigdir}/macros.d/macros.selinux-policy
 %{_unitdir}/selinux-check-proper-disable.service
 %{_libexecdir}/selinux/binsbin-convert.sh
 %{_libexecdir}/selinux/varrun-convert.sh
@@ -157,6 +152,17 @@ and some additional files.
 selinuxenabled && sepolgen-ifgen 2>/dev/null
 exit 0
 
+%package rpm-macros
+Summary: SELinux policy rpm macros
+BuildArch: noarch
+
+%description rpm-macros
+SELinux policy rpm macros package.
+This package contains macros for building SELinux policy modules.
+
+%files rpm-macros
+%{_rpmconfigdir}/macros.d/macros.selinux-policy
+
 %package doc
 Summary: SELinux policy documentation
 Requires(pre): selinux-policy = %{version}-%{release}
@@ -724,6 +730,7 @@ exit 0
 %config(noreplace) %{_sysconfdir}/selinux/minimum/contexts/users/unconfined_u
 %config(noreplace) %{_sysconfdir}/selinux/minimum/contexts/users/sysadm_u
 %fileList minimum
+%dir %{_datadir}/selinux/packages/minimum
 %{_datadir}/selinux/minimum/modules-enabled.lst
 %endif
 
@@ -781,6 +788,7 @@ exit 0
 %config(noreplace) %{_sysconfdir}/dnf/protected.d/selinux-policy-mls.conf
 %config(noreplace) %{_sysconfdir}/selinux/mls/contexts/users/unconfined_u
 %fileList mls
+%dir %{_datadir}/selinux/packages/mls
 %endif
 
 %changelog
