diff --git OLD/shadow-utils.spec NEW/shadow-utils.spec
index ec8df78..a4d4484 100644
--- OLD/shadow-utils.spec
+++ NEW/shadow-utils.spec
@@ -78,7 +78,6 @@ Obsoletes: passwd <= 0.80-19
 # Compat symlinks for Requires in other packages.
 # We rely on filesystem to create the symlinks for us.
 Requires: filesystem(unmerged-sbin-symlinks)
-Provides: /usr/sbin/adduser
 Provides: /usr/sbin/useradd
 Provides: /usr/sbin/userdel
 Provides: /usr/sbin/usermod
@@ -155,13 +154,6 @@ install -p -c -m 0600 %{SOURCE2} $RPM_BUILD_ROOT%{_sysconfdir}/default/useradd
 install -d -m 755 $RPM_BUILD_ROOT%{_pam_confdir}
 install -m 644 %{SOURCE7} $RPM_BUILD_ROOT%{_pam_confdir}/passwd
 
-
-ln -s useradd $RPM_BUILD_ROOT%{_sbindir}/adduser
-ln -s useradd.8 $RPM_BUILD_ROOT%{_mandir}/man8/adduser.8
-for subdir in $RPM_BUILD_ROOT%{_mandir}/{??,??_??,??_??.*}/man* ; do
-        test -d $subdir && test -e $subdir/useradd.8 && echo ".so man8/useradd.8" > $subdir/adduser.8
-done
-
 %if "%{_sbindir}" == "%{_bindir}"
 # The installation script doesn't actually use the configured paths :(
 mv -v $RPM_BUILD_ROOT/usr/sbin/* $RPM_BUILD_ROOT%{_bindir}/
@@ -176,6 +168,8 @@ rm $RPM_BUILD_ROOT%{_bindir}/su
 rm $RPM_BUILD_ROOT%{_bindir}/faillog
 rm $RPM_BUILD_ROOT%{_sbindir}/logoutd
 rm $RPM_BUILD_ROOT%{_sbindir}/nologin
+rm $RPM_BUILD_ROOT%{_sbindir}/vipw
+rm $RPM_BUILD_ROOT%{_sbindir}/vigr
 rm $RPM_BUILD_ROOT%{_mandir}/man1/chfn.*
 rm $RPM_BUILD_ROOT%{_mandir}/*/man1/chfn.*
 rm $RPM_BUILD_ROOT%{_mandir}/man1/chsh.*
@@ -198,6 +192,10 @@ rm $RPM_BUILD_ROOT%{_mandir}/man5/faillog.*
 rm $RPM_BUILD_ROOT%{_mandir}/*/man5/faillog.*
 rm $RPM_BUILD_ROOT%{_mandir}/man8/faillog.*
 rm $RPM_BUILD_ROOT%{_mandir}/*/man8/faillog.*
+rm $RPM_BUILD_ROOT%{_mandir}/man8/vipw.8*
+rm $RPM_BUILD_ROOT%{_mandir}/*/man8/vipw.8*
+rm $RPM_BUILD_ROOT%{_mandir}/man8/vigr.8*
+rm $RPM_BUILD_ROOT%{_mandir}/*/man8/vigr.8*
 
 # Remove PAM service files we don't use.
 rm $RPM_BUILD_ROOT%{_pam_confdir}/chfn
@@ -240,7 +238,6 @@ rm -f $RPM_BUILD_ROOT/%{_libdir}/libsubid.a
 %attr(0755,root,root) %caps(cap_setgid=ep) %{_bindir}/newgidmap
 %attr(0755,root,root) %caps(cap_setuid=ep) %{_bindir}/newuidmap
 %attr(4755,root,root) %{_bindir}/passwd
-%{_sbindir}/adduser
 %attr(0755,root,root)   %{_sbindir}/user*
 %attr(0755,root,root)   %{_sbindir}/group*
 %{_sbindir}/grpck
@@ -249,8 +246,6 @@ rm -f $RPM_BUILD_ROOT/%{_libdir}/libsubid.a
 %{_sbindir}/chpasswd
 %{_sbindir}/chgpasswd
 %{_sbindir}/newusers
-%{_sbindir}/vipw
-%{_sbindir}/vigr
 %{_mandir}/man1/chage.1*
 %{_mandir}/man1/gpasswd.1*
 %{_mandir}/man1/sg.1*
@@ -264,7 +259,6 @@ rm -f $RPM_BUILD_ROOT/%{_libdir}/libsubid.a
 %{_mandir}/man5/gshadow.5*
 %{_mandir}/man5/subuid.5*
 %{_mandir}/man5/subgid.5*
-%{_mandir}/man8/adduser.8*
 %{_mandir}/man8/group*.8*
 %{_mandir}/man8/user*.8*
 %{_mandir}/man8/pwck.8*
@@ -273,8 +267,6 @@ rm -f $RPM_BUILD_ROOT/%{_libdir}/libsubid.a
 %{_mandir}/man8/chgpasswd.8*
 %{_mandir}/man8/newusers.8*
 %{_mandir}/man8/*conv.8*
-%{_mandir}/man8/vipw.8*
-%{_mandir}/man8/vigr.8*
 
 %files subid
 %{_libdir}/libsubid.so.*
