diff --git OLD/switcheroo-control.spec NEW/switcheroo-control.spec
index 8590a41..f6e2a37 100644
--- OLD/switcheroo-control.spec
+++ NEW/switcheroo-control.spec
@@ -8,6 +8,9 @@ URL:            https://gitlab.freedesktop.org/hadess/switcheroo-control/
 # URL from https://gitlab.freedesktop.org/hadess/switcheroo-control/-/releases
 Source0:        https://gitlab.freedesktop.org/hadess/switcheroo-control/-/releases/3.0/downloads/switcheroo-control-3.0.tar.xz
 
+# Replace switcherooctl with a native executable
+Patch0:         switcheroo-control-native-switcherooctl.patch
+
 BuildRequires:  gcc
 BuildRequires:  pkgconfig(gudev-1.0)
 BuildRequires:  pkgconfig(gio-2.0)
@@ -33,7 +36,7 @@ BuildArch:      noarch
 This package contains the documentation for %{name}.
 
 %prep
-%autosetup
+%autosetup -p1
 
 
 %build
