--- systemd.spec	2024-11-29 19:42:01.166625130 +0800
+++ systemd.spec.new	2024-11-29 21:34:27.974572510 +0800
@@ -536,9 +536,8 @@
 the version that works with Secure Boot.
 %endif
 
-%package container
-# Name is the same as in Debian
-Summary: Tools for containers and VMs
+%package machine-common
+Summary: Tools for both nspawn and vmspawn
 Requires:       %{name}%{_isa} = %{version}-%{release}
 Requires(post):   systemd%{_isa} = %{version}-%{release}
 Requires(preun):  systemd%{_isa} = %{version}-%{release}
@@ -549,11 +548,40 @@
 Suggests:       libcurl-minimal
 License:        LGPL-2.1-or-later
 
-%description container
+%description machine-common
 Systemd tools to spawn and manage containers and virtual machines.
 
-This package contains systemd-nspawn, systemd-vmspawn, machinectl,
-systemd-machined, and systemd-importd.
+This package contains machinectl, systemd-machined, and systemd-importd.
+
+%package nspawn
+Summary: Tools for containers
+Requires:       %{name}-machine-common%{_isa} = %{version}-%{release}
+Requires(post):   systemd%{_isa} = %{version}-%{release}
+Requires(preun):  systemd%{_isa} = %{version}-%{release}
+Requires(postun): systemd%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description nspawn
+Systemd tools to spawn and manage containers.
+
+This package contains systemd-nspawn.
+
+%package vmspawn
+Summary: Tools for VMs
+Requires:       %{name}-machine-common%{_isa} = %{version}-%{release}
+Requires(post):   systemd%{_isa} = %{version}-%{release}
+Requires(preun):  systemd%{_isa} = %{version}-%{release}
+Requires(postun): systemd%{_isa} = %{version}-%{release}
+# For systemd-vmspawn which uses qemu:
+Requires:       qemu-kvm
+Recommends:     qemu-device-display-virtio-gpu
+Recommends:     qemu-device-display-virtio-vga
+License:        LGPL-2.1-or-later
+
+%description vmspawn
+Systemd tools to spawn and manage virtual machines.
+
+This package contains systemd-vmspawn.
 
 %package journal-remote
 # Name is the same as in Debian
@@ -1036,6 +1064,10 @@
 # Split files in build root into rpms
 python3 %{SOURCE2} %buildroot %{!?want_bootloader:--no-bootloader}
 
+# Split out the nspawn and vmspawn units
+sed -i -e '/nspawn/ w .file-list-nspawn' -e '/nspawn/ d' .file-list-container
+sed -i -e '/vmspawn/ w .file-list-vmspawn' -e '/vmspawn/ d' .file-list-container
+
 %check
 %if %{with tests}
 meson test -C %{_vpath_builddir} -t 6 --print-errorlogs
@@ -1289,9 +1321,13 @@
 %files boot-unsigned -f .file-list-boot
 %endif
 
-%files container -f .file-list-container
+%files machine-common -f .file-list-container
 %ghost %dir %attr(0700,-,-) /var/lib/machines
 
+%files nspawn -f .file-list-nspawn
+
+%files vmspawn -f .file-list-vmspawn
+
 %files journal-remote -f .file-list-remote
 
 %files networkd -f .file-list-networkd
