--- systemd.spec	2025-02-08 20:48:14.028505220 +0800
+++ systemd.spec.new	2025-04-03 17:38:28.205600129 +0800
@@ -291,6 +291,11 @@
 Conflicts:      %{name}-standalone-shutdown
 Provides:       %{name}-shutdown = %{version}-%{release}
 
+# If we are in a container
+Requires:       (%{name}-containerized = %{version}-%{release} if fedora-release-container)
+# If we are in a atomic or ostree desktop
+Requires:       (%{name}-atomic = %{version}-%{release} if fedora-release-ostree-desktop)
+
 %if "%{_sbindir}" == "%{_bindir}"
 # Compat symlinks for Requires in other packages.
 # We rely on filesystem to create the symlinks for us.
@@ -309,11 +314,6 @@
 Recommends:     libidn2.so.0%{?elf_suffix}
 Recommends:     libidn2.so.0(IDN2_0.0.0)%{?elf_bits}
 Recommends:     libpcre2-8.so.0%{?elf_suffix}
-Recommends:     libpwquality.so.1%{?elf_suffix}
-Recommends:     libpwquality.so.1(LIBPWQUALITY_1.0)%{?elf_bits}
-%if 0%{?fedora}
-Recommends:     libqrencode.so.4%{?elf_suffix}
-%endif
 Recommends:     libbpf.so.1%{?elf_suffix}
 Recommends:     libbpf.so.1(LIBBPF_0.4.0)%{?elf_bits}
 
@@ -477,9 +477,6 @@
 
 This package also provides systemd-timesyncd, a network time protocol daemon.
 
-It also contains tools to manage encrypted home areas and secrets bound to the
-machine, and to create or grow partitions and make file systems automatically.
-
 %package ukify
 Summary:        Tool to build Unified Kernel Images
 Requires:       %{name} = %{version}-%{release}
@@ -536,28 +533,131 @@
 the version that works with Secure Boot.
 %endif
 
-%package container
-# Name is the same as in Debian
-Summary: Tools for containers and VMs
+%package home
+Summary:        Home directory manager
+Requires:       %{name} = %{version}-%{release}
+Requires:       (%{name}-ssh = %{version}-%{release} if openssh-server)
+Recommends:     libpwquality.so.1%{?elf_suffix}
+Recommends:     libpwquality.so.1(LIBPWQUALITY_1.0)%{?elf_bits}
+%if 0%{?fedora}
+Recommends:     libqrencode.so.4%{?elf_suffix}
+%endif
+License:        LGPL-2.1-or-later
+
+%description home
+Systemd tools to manage home directories.
+
+This package contains systemd-homed, which manages home directories in a more
+secure and convenient way.
+
+%package portable
+Summary:        Systemd portable service manager
+Requires:       %{name} = %{version}-%{release}
+Recommends:     %{name}-rootless%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description portable
+This package provides systemd's portable service manager, enabling the management
+of services in a containerized environment.
+
+%package ssh
+Summary:        Systemd SSH and SSHD related tools and configuration
+Requires:       %{name} = %{version}-%{release}
+Recommends:     openssh-clients
+Recommends:     openssh-server
+License:        LGPL-2.1-or-later
+
+%description ssh
+This package provides tools and configurations for SSH clients and servers.
+
+%package containerized
+Summary:        Systemd containerized components
+Requires:       %{name} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description containerized
+This package provides systemd's containerized components, enabling the features
+of systemd in a containerized environment.
+
+%package atomic
+Summary:        Systemd atomic desktop components
+Requires:       %{name} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description atomic
+This package provides systemd's atomic desktop components, enabling the management
+of atomic operations within desktop environments.
+
+%package dissect
+Summary:        Systemd image dissect tool
+Requires:       %{name} = %{version}-%{release}
+Recommends:     %{name}-rootless%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description dissect
+This package provides systemd's dissect tool, enabling the analysis of disk
+images and partitions.
+
+%package machine
+Summary:        Tools for both nspawn and vmspawn
 Requires:       %{name}%{_isa} = %{version}-%{release}
-Requires(post):   systemd%{_isa} = %{version}-%{release}
-Requires(preun):  systemd%{_isa} = %{version}-%{release}
-Requires(postun): systemd%{_isa} = %{version}-%{release}
+Requires:       (%{name}-ssh = %{version}-%{release} if openssh-clients)
 # For systemd-vmspawn which uses qemu:
-Recommends:     qemu-kvm-core
-Recommends:     qemu-device-display-virtio-gpu
-Recommends:     qemu-device-display-virtio-vga
+Requires:       (qemu-kvm-core if %{name}-vmspawn)
+%if 0%{?fedora}
+Recommends:     (qemu-device-display-virtio-gpu if %{name}-vmspawn)
+Recommends:     (qemu-device-display-virtio-vga if %{name}-vmspawn)
+%endif
 # Obsolete parent package so that dnf will install new subpackage on upgrade (#1260394)
 Obsoletes:      %{name} < 229-5
+Recommends:     %{name}-importd%{_isa} = %{version}-%{release}
+Recommends:     (%{name}-rootless%{_isa} = %{version}-%{release} if %{name}-nspawn)
+License:        LGPL-2.1-or-later
+
+%description machine
+Systemd tools to spawn and manage containers and virtual machines.
+
+This package contains machinectl, systemd-machined.
+
+%package importd
+Summary:        Tools to download, import and export container and VM images
+Requires:       %{name}-machine%{_isa} = %{version}-%{release}
 # Bias the system towards libcurl-minimal if nothing pulls in full libcurl (#1997040)
 Suggests:       libcurl-minimal
 License:        LGPL-2.1-or-later
 
-%description container
-Systemd tools to spawn and manage containers and virtual machines.
+%description importd
+Systemd tools to download, import, and export container and VM images.
+
+%package rootless
+Summary:        Daemon to run rootless containers
+Requires:       %{name}%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description rootless
+Systemd daemon to run rootless containers.
 
-This package contains systemd-nspawn, systemd-vmspawn, machinectl,
-systemd-machined, and systemd-importd.
+This package contains systemd-mountfsd and systemd-nsresourced.
+
+%package nspawn
+Summary:        Tools for containers
+Requires:       %{name}-machine%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description nspawn
+Systemd tools to spawn and manage containers.
+
+This package contains systemd-nspawn.
+
+%package vmspawn
+Summary:        Tools for VMs
+Requires:       %{name}-machine%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description vmspawn
+Systemd tools to spawn and manage virtual machines.
+
+This package contains systemd-vmspawn.
 
 %package journal-remote
 # Name is the same as in Debian
@@ -1040,6 +1140,26 @@
 # Split files in build root into rpms
 python3 %{SOURCE2} %buildroot %{!?want_bootloader:--no-bootloader}
 
+# Split out the units
+sed -i -e '/nspawn/ { s/.*/echo "&" >> .file-list-nspawn/e; d; }' .file-list-container
+sed -i -e '/vmspawn/ { s/.*/echo "&" >> .file-list-vmspawn/e; d; }' .file-list-container
+sed -i -e '/import\|pull\|export/ { s/.*/echo "&" >> .file-list-importd/e; d; }' .file-list-container
+sed -i -e '/container/ { s/.*/echo "&" >> .file-list-containerized/e; d; }' .file-list-main
+sed -i -e '/confext\|sysext\|sysupdate\|extension/ { s/.*/echo "&" >> .file-list-atomic/e; d; }' .file-list-main
+sed -i -e '/confext\|sysext/ { s/.*/echo "&" >> .file-list-atomic/e; d; }' .file-list-udev
+sed -i -e '/dissect/ { s/.*/echo "&" >> .file-list-dissect/e; d; }' .file-list-container
+sed -i -e '/ddi/ { s/.*/echo "&" >> .file-list-dissect/e; d; }' .file-list-main
+sed -i -e '/mountfs\|mountwork\|nsresource/ { s/.*/echo "&" >> .file-list-rootless/e; d; }' .file-list-container
+sed -i -e '/ssh/ { s/.*/echo "&" >> .file-list-ssh/e; d; }' .file-list-main
+sed -i -e '/home/ { s/.*/echo "&" >> .file-list-home/e; d; }' .file-list-main
+sed -i -e '/home/ { s/.*/echo "&" >> .file-list-home/e; d; }' .file-list-udev
+sed -i -e '/portable/ { s/.*/echo "&" >> .file-list-portable/e; d; }' .file-list-main
+sed -i -e '/portable/ { s/.*/echo "&" >> .file-list-portable/e; d; }' .file-list-container
+sed -i -e '/portable/ { s/.*/echo "&" >> .file-list-portable/e; d; }' .file-list-udev
+sed -i -e '/resolv/ { s/.*/echo "&" >> .file-list-resolve/e; d; }' .file-list-main
+sed -i -e '/network-generator/ { s/.*/echo "&" >> .file-list-networkd/e; d; }' .file-list-main
+sed -i -e '/fstab\|tpm\|battery\|bluetooth\|printer\|smartcard\|modprobe\|wait\|ntp\|kernel/ { s/.*/echo "&" >> .file-list-udev/e; d; }' .file-list-main
+
 %check
 %if %{with tests}
 meson test -C %{_vpath_builddir} -t 6 --print-errorlogs
@@ -1103,7 +1223,7 @@
 # a different package version.
 systemctl --no-reload preset systemd-journald-audit.socket &>/dev/null || :
 
-%global udev_services systemd-udev{d,-settle,-trigger}.service systemd-udevd-{control,kernel}.socket systemd-homed.service %{?want_bootloader:systemd-boot-update.service} systemd-oomd.service systemd-portabled.service systemd-pstore.service systemd-timesyncd.service remote-cryptsetup.target
+%global udev_services systemd-udev{d,-settle,-trigger}.service systemd-udevd-{control,kernel}.socket %{?want_bootloader:systemd-boot-update.service} systemd-oomd.service systemd-pstore.service systemd-timesyncd.service remote-cryptsetup.target
 
 %post udev
 # Move old stuff around in /var/lib
@@ -1139,6 +1259,17 @@
 # Others are either oneshot services, or sockets, and restarting them causes issues (#1378974)
 %systemd_postun_with_restart systemd-udevd.service systemd-timesyncd.service
 
+%post home
+%systemd_post systemd-homed.service
+
+%preun home
+%systemd_preun systemd-homed.service
+
+%post portable
+%systemd_post systemd-portabled.service
+
+%preun portable
+%systemd_preun systemd-portabled.service
 
 %global journal_remote_units_restart systemd-journal-gatewayd.service systemd-journal-remote.service systemd-journal-upload.service
 %global journal_remote_units_norestart systemd-journal-gatewayd.socket systemd-journal-remote.socket
@@ -1256,21 +1387,17 @@
 %license LICENSE.GPL2
 %license LICENSES/MIT.txt
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/basic.target.wants
-%ghost %dir %attr(0755,-,-) /etc/systemd/system/bluetooth.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/default.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/getty.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/graphical.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/local-fs.target.wants
-%ghost %dir %attr(0755,-,-) /etc/systemd/system/machines.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/multi-user.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/network-online.target.wants
-%ghost %dir %attr(0755,-,-) /etc/systemd/system/printer.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/remote-fs.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/sockets.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/sysinit.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/system-update.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/timers.target.wants
-%ghost %dir %attr(0700,-,-) /var/lib/portables
 %ghost %dir %attr(0755,-,-) /var/lib/rpm-state/systemd
 
 %files libs -f .file-list-libs
@@ -1285,15 +1412,39 @@
 %files devel -f .file-list-devel
 
 %files udev -f .file-list-udev
+%ghost %dir %attr(0755,-,-) /etc/systemd/system/bluetooth.target.wants
+%ghost %dir %attr(0755,-,-) /etc/systemd/system/printer.target.wants
 
 %files ukify -f .file-list-ukify
 %if 0%{?want_bootloader}
 %files boot-unsigned -f .file-list-boot
 %endif
 
-%files container -f .file-list-container
+%files home -f .file-list-home
+
+%files portable -f .file-list-portable
+%ghost %dir %attr(0700,-,-) /var/lib/portables
+
+%files ssh -f .file-list-ssh
+
+%files containerized -f .file-list-containerized
+
+%files atomic -f .file-list-atomic
+
+%files dissect -f .file-list-dissect
+
+%files machine -f .file-list-container
+%ghost %dir %attr(0755,-,-) /etc/systemd/system/machines.target.wants
 %ghost %dir %attr(0700,-,-) /var/lib/machines
 
+%files importd -f .file-list-importd
+
+%files rootless -f .file-list-rootless
+
+%files nspawn -f .file-list-nspawn
+
+%files vmspawn -f .file-list-vmspawn
+
 %files journal-remote -f .file-list-remote
 
 %files networkd -f .file-list-networkd
