diff --git OLD/systemd.spec NEW/systemd.spec
index 517ee20..985a2ee 100644
--- OLD/systemd.spec
+++ NEW/systemd.spec
@@ -294,10 +294,6 @@ Provides:       syslog
 Provides:       systemd-units = %{version}-%{release}
 Obsoletes:      system-setup-keyboard < 0.9
 Provides:       system-setup-keyboard = 0.9
-# systemd-sysv-convert was removed in f20: https://fedorahosted.org/fpc/ticket/308
-Obsoletes:      systemd-sysv < 206
-Provides:       systemd-sysv = 206
-Conflicts:      initscripts < 9.56.1
 %if 0%{?fedora}
 Conflicts:      fedora-release < 23-0.12
 %endif
@@ -322,6 +318,14 @@ Provides:       %{name}-tmpfiles = %{version}-%{release}
 Conflicts:      %{name}-standalone-shutdown
 Provides:       %{name}-shutdown = %{version}-%{release}
 
+# If we are in a container
+Requires:       (%{name}-containerized = %{version}-%{release} if fedora-release-container)
+# If we are in a atomic or ostree desktop
+Requires:       (%{name}-atomic = %{version}-%{release} if %{shrink:(
+        fedora-release-ostree-desktop or
+        fedora-release-coreos
+)})
+
 %if "%{_sbindir}" == "%{_bindir}"
 # Compat symlinks for Requires in other packages.
 # We rely on filesystem to create the symlinks for us.
@@ -340,11 +344,6 @@ Recommends:     libxkbcommon.so.0%{?elf_suffix}
 Recommends:     libidn2.so.0%{?elf_suffix}
 Recommends:     libidn2.so.0(IDN2_0.0.0)%{?elf_bits}
 Recommends:     libpcre2-8.so.0%{?elf_suffix}
-Recommends:     libpwquality.so.1%{?elf_suffix}
-Recommends:     libpwquality.so.1(LIBPWQUALITY_1.0)%{?elf_bits}
-%if 0%{?fedora}
-Recommends:     libqrencode.so.4%{?elf_suffix}
-%endif
 Recommends:     libbpf.so.1%{?elf_suffix}
 Recommends:     libbpf.so.1(LIBBPF_0.4.0)%{?elf_bits}
 
@@ -358,15 +357,6 @@ Recommends:     libelf.so.1(ELFUTILS_1.7)%{?elf_bits}
 Recommends:     libcryptsetup.so.12%{?elf_suffix}
 Recommends:     libcryptsetup.so.12(CRYPTSETUP_2.4)%{?elf_bits}
 
-# Libkmod is used to load modules.
-Recommends:     libkmod.so.2%{?elf_suffix}
-# kmod_list_next, kmod_load_resources, kmod_module_get_initstate,
-# kmod_module_get_module, kmod_module_get_name, kmod_module_new_from_lookup,
-# kmod_module_probe_insert_module, kmod_module_unref, kmod_module_unref_list,
-# kmod_new, kmod_set_log_fn, kmod_unref, kmod_validate_resources
-# are part of LIBKMOD_5.
-Recommends:     libkmod.so.2(LIBKMOD_5)%{?elf_bits}
-
 Recommends:     libarchive.so.13%{?elf_suffix}
 
 %description
@@ -375,8 +365,7 @@ of the system. It provides aggressive parallelization capabilities, uses socket
 and D-Bus activation for starting services, offers on-demand starting of
 daemons, keeps track of processes using Linux control groups, maintains mount
 and automount points, and implements an elaborate transactional dependency-based
-service control logic. systemd supports SysV and LSB init scripts and works as a
-replacement for sysvinit. Other parts of this package are a logging daemon,
+service control logic. Other parts of this package are a logging daemon,
 utilities to control basic system configuration like the hostname, date, locale,
 maintain a list of logged-in users, system accounts, runtime directories and
 settings, and a logging daemons.
@@ -429,7 +418,7 @@ This package contains the systemd-sysusers program.
 %package devel
 Summary:        Development headers for systemd
 License:        LGPL-2.1-or-later AND MIT
-Requires:       %{name}-libs%{_isa} = %{version}-%{release}
+Requires:       %{name}%{_isa} = %{version}-%{release}
 Requires(meta): (%{name}-rpm-macros = %{version}-%{release} if rpm-build)
 Provides:       libudev-devel = %{version}
 Provides:       libudev-devel%{_isa} = %{version}
@@ -513,9 +502,6 @@ virtual machines, but not in containers.
 
 This package also provides systemd-timesyncd, a network time protocol daemon.
 
-It also contains tools to manage encrypted home areas and secrets bound to the
-machine, and to create or grow partitions and make file systems automatically.
-
 %package ukify
 Summary:        Tool to build Unified Kernel Images
 Requires:       %{name} = %{noarch_requires_version}
@@ -588,28 +574,129 @@ This package contains the signed version.
 %endif
 %endif
 
-%package container
-# Name is the same as in Debian
-Summary: Tools for containers and VMs
+%package home
+Summary:        Home directory manager
+Requires:       %{name} = %{version}-%{release}
+Requires:       (%{name}-ssh = %{version}-%{release} if openssh-server)
+Recommends:     libpwquality.so.1%{?elf_suffix}
+Recommends:     libpwquality.so.1(LIBPWQUALITY_1.0)%{?elf_bits}
+%if 0%{?fedora}
+Recommends:     libqrencode.so.4%{?elf_suffix}
+%endif
+License:        LGPL-2.1-or-later
+
+%description home
+Systemd tools to manage home directories.
+
+This package contains systemd-homed, which manages home directories in a more
+secure and convenient way.
+
+%package portable
+Summary:        Systemd portable service manager
+Requires:       %{name} = %{version}-%{release}
+Recommends:     %{name}-rootless%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description portable
+This package provides systemd's portable service manager, enabling the management
+of services in a containerized environment.
+
+%package ssh
+Summary:        Systemd SSH and SSHD related tools and configuration
+Requires:       %{name} = %{version}-%{release}
+Recommends:     openssh-clients
+Recommends:     openssh-server
+License:        LGPL-2.1-or-later
+
+%description ssh
+This package provides tools and configurations for SSH clients and servers.
+
+%package containerized
+Summary:        Systemd containerized components
+Requires:       %{name} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description containerized
+This package provides systemd's containerized components, enabling the features
+of systemd in a containerized environment.
+
+%package atomic
+Summary:        Systemd atomic desktop components
+Requires:       %{name} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description atomic
+This package provides systemd's atomic desktop components, enabling the management
+of atomic operations within desktop environments.
+
+%package machine
+Summary:        Tools for both nspawn and vmspawn
 Requires:       %{name}%{_isa} = %{version}-%{release}
-Requires(post):   systemd%{_isa} = %{version}-%{release}
-Requires(preun):  systemd%{_isa} = %{version}-%{release}
-Requires(postun): systemd%{_isa} = %{version}-%{release}
+Requires:       (%{name}-ssh = %{version}-%{release} if openssh-clients)
 # For systemd-vmspawn which uses qemu:
-Recommends:     qemu-kvm-core
+Requires:       (qemu-kvm-core if %{name}-vmspawn)
 %if 0%{?fedora}
-Recommends:     qemu-device-display-virtio-gpu
-Recommends:     qemu-device-display-virtio-vga
+Recommends:     (qemu-device-display-virtio-gpu if %{name}-vmspawn)
+Recommends:     (qemu-device-display-virtio-vga if %{name}-vmspawn)
 %endif
+Recommends:     %{name}-import%{_isa} = %{version}-%{release}
+Recommends:     (%{name}-rootless%{_isa} = %{version}-%{release} if %{name}-nspawn)
+License:        LGPL-2.1-or-later
+
+%description machine
+Systemd tools to spawn and manage containers and virtual machines.
+
+This package contains machinectl, systemd-machined.
+
+%package import
+Summary:        Tools to download, import and export container and VM images
+Requires:       %{name}-machine%{_isa} = %{version}-%{release}
 # Bias the system towards libcurl-minimal if nothing pulls in full libcurl (#1997040)
 Suggests:       libcurl-minimal
 License:        LGPL-2.1-or-later
 
-%description container
-Systemd tools to spawn and manage containers and virtual machines.
+%description import
+Systemd tools to download, import, and export container and VM images.
+
+%package rootless
+Summary:        Daemon to run rootless containers
+Requires:       %{name}%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description rootless
+Systemd daemon to run rootless containers.
+
+This package contains systemd-mountfsd and systemd-nsresourced.
+
+%package nspawn
+Summary:        Tools for containers
+Requires:       %{name}-machine%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description nspawn
+Systemd tools to spawn and manage containers.
+
+This package contains systemd-nspawn.
 
-This package contains systemd-nspawn, systemd-vmspawn, machinectl,
-systemd-machined, and systemd-importd.
+%package vmspawn
+Summary:        Tools for VMs
+Requires:       %{name}-machine%{_isa} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description vmspawn
+Systemd tools to spawn and manage virtual machines.
+
+This package contains systemd-vmspawn.
+
+%package storagetm
+Summary:        Systemd storage target mode
+Requires:       %{name} = %{version}-%{release}
+License:        LGPL-2.1-or-later
+
+%description storagetm
+Systemd storage target mode.
+
+This package contains systemd-storagetm.
 
 %package journal-remote
 # Name is the same as in Debian
@@ -776,8 +863,8 @@ VMLINUX_H_PATH=$(%python3 -c '%find_vmlinux_h')
 CONFIGURE_OPTS=(
         -Dmode=release
         -Dslow-tests=true
-        -Dsysvinit-path=/etc/rc.d/init.d
-        -Drc-local=/etc/rc.d/rc.local
+        -Dsysvinit-path=
+        -Dsysvrcnd-path=
         -Dntp-servers='0.%{ntpvendor}.pool.ntp.org 1.%{ntpvendor}.pool.ntp.org 2.%{ntpvendor}.pool.ntp.org 3.%{ntpvendor}.pool.ntp.org'
         -Ddns-servers=
         -Dservice-watchdog=
@@ -1120,6 +1207,30 @@ python3 %{SOURCE2} %buildroot %{!?want_bootloader:--no-bootloader}
 BRP_PESIGN_FILES=/usr/lib/systemd/boot/efi/systemd-boot%{efi_arch}.efi BRP_PESIGN_PACKAGES=systemd-boot /usr/lib/rpm/brp-suse.d/brp-99-pesign
 %endif
 
+# Split out the units
+sed -i -e '/nspawn/ { s/.*/echo "&" >> .file-list-nspawn/e; d; }' .file-list-container
+sed -i -e '/vmspawn/ { s/.*/echo "&" >> .file-list-vmspawn/e; d; }' .file-list-container
+sed -i -e '/import\|pull\|export\|dissect\|ddi/ { s/.*/echo "&" >> .file-list-import/e; d; }' .file-list-container
+sed -i -e '/container/ { s/.*/echo "&" >> .file-list-containerized/e; d; }' .file-list-main
+sed -i -e '/confext\|sysext\|sysupdate\|extension/ { s/.*/echo "&" >> .file-list-atomic/e; d; }' .file-list-main
+sed -i -e '/confext\|sysext/ { s/.*/echo "&" >> .file-list-atomic/e; d; }' .file-list-udev
+sed -i -e '/mountfs\|mountwork\|nsresource/ { s/.*/echo "&" >> .file-list-rootless/e; d; }' .file-list-container
+sed -i -e '/ssh/ { s/.*/echo "&" >> .file-list-ssh/e; d; }' .file-list-main
+sed -i -e '/home/ { s/.*/echo "&" >> .file-list-home/e; d; }' .file-list-main
+sed -i -e '/home/ { s/.*/echo "&" >> .file-list-home/e; d; }' .file-list-udev
+sed -i -e '/portable/ { s/.*/echo "&" >> .file-list-portable/e; d; }' .file-list-main
+sed -i -e '/portable/ { s/.*/echo "&" >> .file-list-portable/e; d; }' .file-list-container
+sed -i -e '/portable/ { s/.*/echo "&" >> .file-list-portable/e; d; }' .file-list-udev
+sed -i -e '/resolv/ { s/.*/echo "&" >> .file-list-resolve/e; d; }' .file-list-main
+sed -i -e '/resolve/ { s/.*/echo "&" >> .file-list-resolve/e; d; }' .file-list-libs
+sed -i -e '/network-generator/ { s/.*/echo "&" >> .file-list-networkd/e; d; }' .file-list-udev
+sed -i -e '/fstab\|tpm\|battery\|bluetooth\|printer\|smartcard\|modprobe\|wait\|ntp\|kernel\|pcrlock\|emergency\|rescue\|getty\|sigpwr\|usb-gadget\|time-sync\|debug/ { s/.*/echo "&" >> .file-list-udev/e; d; }' .file-list-main
+sed -i -e '/stub\|sbsign/ { s/.*/echo "&" >> .file-list-boot/e; d; }' .file-list-main
+sed -i -e '/bootctl/ { s/.*/echo "&" >> .file-list-boot/e; d; }' .file-list-udev
+sed -i -e '/pkgconfig/ { s/.*/echo "&" >> .file-list-devel/e; d; }' .file-list-main
+sed -i -e '/storage/ { s/.*/echo "&" >> .file-list-storagetm/e; d; }' .file-list-main
+sed -i -e '/libnss/ { s/.*/echo "&" >> .file-list-main/e; d; }' .file-list-libs
+
 %check
 %if %{with tests}
 meson test -C %{_vpath_builddir} -t 6 --print-errorlogs
@@ -1237,9 +1348,6 @@ systemctl --no-reload preset systemd-journald-audit.socket &>/dev/null || :
                         systemd-hibernate-clear.service
                         systemd-hibernate-resume.service
                         systemd-hibernate.service
-                        systemd-homed-activate.service
-                        systemd-homed-firstboot.service
-                        systemd-homed.service
                         systemd-hwdb-update.service
                         systemd-hybrid-sleep.service
                         systemd-modules-load.service
@@ -1263,7 +1371,6 @@ systemctl --no-reload preset systemd-journald-audit.socket &>/dev/null || :
                         systemd-pcrphase-initrd.service
                         systemd-pcrphase-sysinit.service
                         systemd-pcrphase.service
-                        systemd-portabled.service
                         systemd-pstore.service
                         systemd-quotacheck-root.service
                         systemd-quotacheck@.service
@@ -1328,7 +1435,19 @@ grep -q -E '^KEYMAP="?fi-latin[19]"?' /etc/vconsole.conf 2>/dev/null &&
 %posttrans udev
 # Restart some services.
 # Others are either oneshot services, or sockets, and restarting them causes issues (#1378974)
-%systemd_posttrans_with_restart systemd-udevd.service systemd-timesyncd.service systemd-homed.service systemd-oomd.service systemd-portabled.service
+%systemd_posttrans_with_restart systemd-udevd.service systemd-timesyncd.service systemd-oomd.service
+
+%preun home
+%systemd_preun systemd-homed-activate.service systemd-homed-firstboot.service systemd-homed.service
+
+%posttrans home
+%systemd_posttrans_with_restart systemd-homed.service
+
+%preun portable
+%systemd_preun systemd-portabled.service
+
+%posttrans portable
+%systemd_posttrans_with_restart systemd-portabled.service
 
 %global journal_remote_units_restart systemd-journal-gatewayd.service systemd-journal-remote.service systemd-journal-upload.service
 %global journal_remote_units_norestart systemd-journal-gatewayd.socket systemd-journal-remote.socket
@@ -1451,21 +1570,16 @@ fi
 %license LICENSE.GPL2
 %license LICENSES/MIT.txt
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/basic.target.wants
-%ghost %dir %attr(0755,-,-) /etc/systemd/system/bluetooth.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/default.target.wants
-%ghost %dir %attr(0755,-,-) /etc/systemd/system/getty.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/graphical.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/local-fs.target.wants
-%ghost %dir %attr(0755,-,-) /etc/systemd/system/machines.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/multi-user.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/network-online.target.wants
-%ghost %dir %attr(0755,-,-) /etc/systemd/system/printer.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/remote-fs.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/sockets.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/sysinit.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/system-update.target.wants
 %ghost %dir %attr(0755,-,-) /etc/systemd/system/timers.target.wants
-%ghost %dir %attr(0700,-,-) /var/lib/portables
 %ghost %dir %attr(0755,-,-) /var/lib/rpm-state/systemd
 
 %files libs -f .file-list-libs
@@ -1486,6 +1600,9 @@ fi
 %files devel -f .file-list-devel
 
 %files udev -f .file-list-udev
+%ghost %dir %attr(0755,-,-) /etc/systemd/system/bluetooth.target.wants
+%ghost %dir %attr(0755,-,-) /etc/systemd/system/getty.target.wants
+%ghost %dir %attr(0755,-,-) /etc/systemd/system/printer.target.wants
 
 %files ukify -f .file-list-ukify
 %if 0%{?want_bootloader}
@@ -1496,9 +1613,31 @@ fi
 %endif
 %endif
 
-%files container -f .file-list-container
+%files home -f .file-list-home
+
+%files portable -f .file-list-portable
+%ghost %dir %attr(0700,-,-) /var/lib/portables
+
+%files ssh -f .file-list-ssh
+
+%files containerized -f .file-list-containerized
+
+%files atomic -f .file-list-atomic
+
+%files machine -f .file-list-container
+%ghost %dir %attr(0755,-,-) /etc/systemd/system/machines.target.wants
 %ghost %dir %attr(0700,-,-) /var/lib/machines
 
+%files import -f .file-list-import
+
+%files rootless -f .file-list-rootless
+
+%files nspawn -f .file-list-nspawn
+
+%files vmspawn -f .file-list-vmspawn
+
+%files storagetm -f .file-list-storagetm
+
 %files journal-remote -f .file-list-remote
 
 %files networkd -f .file-list-networkd
