diff --git OLD/tinysparql.spec NEW/tinysparql.spec
index f66023c..fe6bcdd 100644
--- OLD/tinysparql.spec
+++ NEW/tinysparql.spec
@@ -4,6 +4,8 @@
 
 %global tracker_obsoletes_version 3.8
 
+%bcond dnssd 0
+
 %if 0%{?rhel}
 %bcond libstemmer 0
 %else
@@ -30,7 +32,9 @@ BuildRequires:  meson
 BuildRequires:  python3-gobject
 BuildRequires:  systemd-rpm-macros
 BuildRequires:  vala
+%if %{with dnssd}
 BuildRequires:  pkgconfig(avahi-glib)
+%endif
 BuildRequires:  pkgconfig(dbus-1)
 BuildRequires:  pkgconfig(gobject-introspection-1.0)
 BuildRequires:  pkgconfig(icu-i18n)
@@ -111,6 +115,9 @@ This package contains the documentation for %{name}.
   -Dsystemd_user_services_dir=%{_userunitdir} \
 %if %{without libstemmer}
   -Dstemmer=disabled \
+%endif
+%if %{without dnssd}
+  -Davahi=disabled \
 %endif
   %{nil}
 
