diff --git OLD/tinysparql.spec NEW/tinysparql.spec
index b79687a..e3297d4 100644
--- OLD/tinysparql.spec
+++ NEW/tinysparql.spec
@@ -146,7 +146,6 @@ This package contains the documentation for %{name}.
 %{_libdir}/libtinysparql-3.0.so.0*
 %{_libdir}/libtracker-sparql-3.0.so.0*
 %{_libdir}/tinysparql-3.0/
-%{_datadir}/gir-1.0/Tracker-3.0.gir
 
 %files devel
 %license COPYING COPYING.LGPL
@@ -157,6 +156,7 @@ This package contains the documentation for %{name}.
 %{_libdir}/pkgconfig/tracker-sparql-3.0.pc
 %{_datadir}/vala/vapi/tinysparql-3.0.deps
 %{_datadir}/vala/vapi/tinysparql-3.0.vapi
+%{_datadir}/gir-1.0/Tracker-3.0.gir
 %{_datadir}/gir-1.0/Tsparql-3.0.gir
 %{_datadir}/vala/vapi/tracker-sparql-3.0.deps
 %{_datadir}/vala/vapi/tracker-sparql-3.0.vapi
