diff --git OLD/tinysparql.spec NEW/tinysparql.spec
index b79687a..3d1f6ff 100644
--- OLD/tinysparql.spec
+++ NEW/tinysparql.spec
@@ -25,7 +25,6 @@ BuildRequires:  meson
 BuildRequires:  python3-gobject
 BuildRequires:  systemd-rpm-macros
 BuildRequires:  vala
-BuildRequires:  pkgconfig(avahi-glib)
 BuildRequires:  pkgconfig(dbus-1)
 BuildRequires:  pkgconfig(gobject-introspection-1.0)
 BuildRequires:  pkgconfig(icu-i18n)
@@ -107,6 +106,7 @@ This package contains the documentation for %{name}.
 %if ! 0%{?rhel} || 0%{?rhel} >= 10
   -Dstemmer=disabled \
 %endif
+  -Davahi=disabled \
   %{nil}
 
 %meson_build
@@ -146,7 +146,6 @@ This package contains the documentation for %{name}.
 %{_libdir}/libtinysparql-3.0.so.0*
 %{_libdir}/libtracker-sparql-3.0.so.0*
 %{_libdir}/tinysparql-3.0/
-%{_datadir}/gir-1.0/Tracker-3.0.gir
 
 %files devel
 %license COPYING COPYING.LGPL
@@ -157,6 +156,7 @@ This package contains the documentation for %{name}.
 %{_libdir}/pkgconfig/tracker-sparql-3.0.pc
 %{_datadir}/vala/vapi/tinysparql-3.0.deps
 %{_datadir}/vala/vapi/tinysparql-3.0.vapi
+%{_datadir}/gir-1.0/Tracker-3.0.gir
 %{_datadir}/gir-1.0/Tsparql-3.0.gir
 %{_datadir}/vala/vapi/tracker-sparql-3.0.deps
 %{_datadir}/vala/vapi/tracker-sparql-3.0.vapi
