diff --git OLD/udisks2.spec NEW/udisks2.spec
index d086c17..297b284 100644
--- OLD/udisks2.spec
+++ NEW/udisks2.spec
@@ -105,6 +105,7 @@ Recommends: ntfs-3g
 Recommends: btrfs-progs
 %endif
 
+Requires: (%{name}-zsh-completion = %{version}-%{release} if zsh)
 
 # For /proc/self/mountinfo, only available in 2.6.26 or higher
 Conflicts: kernel < 2.6.26
@@ -191,6 +192,16 @@ Obsoletes: storaged-lsm < %{version}-%{release}
 This package contains module for LSM configuration.
 %endif
 
+%package zsh-completion
+Summary: Zsh shell completion for %{name}
+Requires: %{name} = %{version}-%{release}
+Requires: zsh
+
+BuildArch: noarch
+
+%description zsh-completion
+Zsh shell completion for %{name}
+
 %prep
 %autosetup -p1 -n udisks-%{version}
 rm -f src/tests/dbus-tests/config_h.py
@@ -265,7 +276,6 @@ fi
 
 %{_datadir}/dbus-1/system.d/org.freedesktop.UDisks2.conf
 %{_datadir}/bash-completion/completions/udisksctl
-%{_datadir}/zsh/site-functions/_udisks2
 %{_tmpfilesdir}/%{name}.conf
 %{_unitdir}/udisks2.service
 %{_udevrulesdir}/80-udisks2.rules
@@ -339,6 +349,9 @@ fi
 %attr(0600,root,root) %{_sysconfdir}/udisks2/modules.conf.d/udisks2_lsm.conf
 %endif
 
+%files zsh-completion
+%{_datadir}/zsh/site-functions/*
+
 %changelog
 * Thu Nov 06 2025 Tomas Bzatek <tbzatek@redhat.com> - 2.11.0-1
 - Version 2.11.0
