diff --git OLD/upower.spec NEW/upower.spec
index c436890..bdc09dc 100644
--- OLD/upower.spec
+++ NEW/upower.spec
@@ -26,10 +26,22 @@ BuildRequires:  systemd
 Requires:       %{name}-libs%{?_isa} = %{version}-%{release}
 Requires:       udev
 
+Requires:       (%{name}-zsh-completion = %{version}-%{release} if zsh)
+
 %description
 UPower (formerly DeviceKit-power) provides a daemon, API and command
 line tools for managing power devices attached to the system.
 
+%package zsh-completion
+Summary: Zsh shell completion for %{name}
+Requires: %{name} = %{version}-%{release}
+Requires: zsh
+
+BuildArch: noarch
+
+%description zsh-completion
+Zsh shell completion for %{name}.
+
 %package libs
 Summary: Client libraries for UPower
 Requires: gobject-introspection
@@ -112,7 +124,9 @@ mv $RPM_BUILD_ROOT%{_libexecdir}/upower $RPM_BUILD_ROOT%{_libexecdir}/installed-
 %{_unitdir}/*.service
 %{_datadir}/polkit-1/actions/org.freedesktop.upower.policy
 %{_datadir}/polkit-1/rules.d/org.freedesktop.upower.rules
-%{_datadir}/zsh/*
+
+%files zsh-completion
+%{_datadir}/zsh/site-functions/*
 
 %files libs
 %license COPYING
