diff --git OLD/wpa_supplicant.spec NEW/wpa_supplicant.spec
index 243440c..927118b 100644
--- OLD/wpa_supplicant.spec
+++ NEW/wpa_supplicant.spec
@@ -67,7 +67,6 @@ BuildRequires: gcc
 %if 0%{?fedora} >= 41
 BuildRequires: openssl-devel-engine
 %endif
-Requires(post): systemd-sysv
 Requires(post): systemd
 Requires(preun): systemd
 Requires(postun): systemd
@@ -100,6 +99,12 @@ Graphical User Interface for wpa_supplicant written using QT
 %endif
 
 
+%package examples
+Summary: Example configurations for %{name}
+
+%description examples
+Example configurations for %{name}
+
 %prep
 %autosetup -p1 -n %{name}-%{version}
 
@@ -160,10 +165,13 @@ install -d %{buildroot}%{_mandir}/man{5,8}
 install -m 0644 wpa_supplicant/doc/docbook/*.8 %{buildroot}%{_mandir}/man8
 install -m 0644 wpa_supplicant/doc/docbook/*.5 %{buildroot}%{_mandir}/man5
 
-# some cleanup in docs and examples
+# examples
+chmod -R 0644 wpa_supplicant/examples/*.py
+cp -a wpa_supplicant/examples %{buildroot}/%{_datadir}/%{name}/
+
+# some cleanup in docs
 rm -f  wpa_supplicant/doc/.cvsignore
 rm -rf wpa_supplicant/doc/docbook
-chmod -R 0644 wpa_supplicant/examples/*.py
 
 
 %post
@@ -173,16 +181,6 @@ chmod -R 0644 wpa_supplicant/examples/*.py
 %preun
 %systemd_preun wpa_supplicant.service
 
-%triggerun -- wpa_supplicant < 0.7.3-10
-# Save the current service runlevel info
-# User must manually run systemd-sysv-convert --apply wpa_supplicant
-# to migrate them to systemd targets
-/usr/bin/systemd-sysv-convert --save wpa_supplicant >/dev/null 2>&1 ||:
-
-# Run these because the SysV package being removed won't do them
-/sbin/chkconfig --del wpa_supplicant >/dev/null 2>&1 || :
-/bin/systemctl try-restart wpa_supplicant.service >/dev/null 2>&1 || :
-
 
 %files
 %config(noreplace) %{_sysconfdir}/wpa_supplicant/wpa_supplicant.conf
@@ -208,8 +206,6 @@ chmod -R 0644 wpa_supplicant/examples/*.py
 %doc wpa_supplicant/ChangeLog
 %doc wpa_supplicant/eap_testing.txt
 %doc wpa_supplicant/todo.txt
-%doc wpa_supplicant/wpa_supplicant.conf
-%doc wpa_supplicant/examples
 %license COPYING
 
 
@@ -220,6 +216,11 @@ chmod -R 0644 wpa_supplicant/examples/*.py
 %endif
 
 
+%files examples
+%doc wpa_supplicant/wpa_supplicant.conf
+%{_datadir}/wpa_supplicant
+
+
 %changelog
 * Thu Feb 12 2026 Beniamino Galvani <bgalvani@redhat.com> - 1:2.11-10
 - Fix loading multiple certificates from the ca_cert blob (#2439303)
