diff --git OLD/wsdd.spec NEW/wsdd.spec
index f75cc33..d199f87 100644
--- OLD/wsdd.spec
+++ NEW/wsdd.spec
@@ -10,6 +10,7 @@ Patch:          Modify-systemd-service-for-Fedora.patch
 
 BuildArch:      noarch
 BuildRequires:  systemd
+Requires:       (%{name}-firewalld = %{version}-%{release} if firewalld)
 
 
 %description
@@ -18,6 +19,17 @@ hosts, like your local NAS device, to be found by Web Service Discovery Clients
 like Windows.
 
 
+%package firewalld
+Summary: Web Services Dynamic Discovery host daemon firewalld rules
+Requires: %{name} = %{version}-%{release}
+Requires: firewalld
+
+BuildArch: noarch
+
+%description firewalld
+Web Services Dynamic Discovery host daemon firewalld rules
+
+
 %prep
 %autosetup -p1
 
@@ -50,8 +62,6 @@ install -m0644 -D wsdd.sysusers.conf %{buildroot}%{_sysusersdir}/wsdd.conf
 
 %files
 %{_unitdir}/wsdd.service
-%{_usr}/lib/firewalld/services/wsdd.xml
-%{_usr}/lib/firewalld/services/wsdd-http.xml
 %config(noreplace) %{_sysconfdir}/sysconfig/wsdd
 %{_bindir}/wsdd
 %{_mandir}/man8/wsdd.8*
@@ -59,6 +69,9 @@ install -m0644 -D wsdd.sysusers.conf %{buildroot}%{_sysusersdir}/wsdd.conf
 %doc AUTHORS README.md
 %{_sysusersdir}/wsdd.conf
 
+%files firewalld
+%{_usr}/lib/firewalld/services/wsdd.xml
+%{_usr}/lib/firewalld/services/wsdd-http.xml
 
 %changelog
 * Sat Jan 17 2026 Fedora Release Engineering <releng@fedoraproject.org> - 0.8-6
