diff --git OLD/wsdd.spec NEW/wsdd.spec
index 5013e49..4e9fd25 100644
--- OLD/wsdd.spec
+++ NEW/wsdd.spec
@@ -10,6 +10,7 @@ Patch:          Modify-systemd-service-for-Fedora.patch
 
 BuildArch:      noarch
 BuildRequires:  systemd
+Requires:       (%{name}-firewalld%{?_isa} = %{version}-%{release} if firewalld)
 Requires(pre):  shadow-utils
 
 
@@ -19,6 +20,15 @@ hosts, like your local NAS device, to be found by Web Service Discovery Clients
 like Windows.
 
 
+%package firewalld
+Summary: Web Services Dynamic Discovery host daemon firewalld rules
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: firewalld
+
+%description firewalld
+Web Services Dynamic Discovery host daemon firewalld rules
+
+
 %prep
 %autosetup -p1
 
@@ -50,14 +60,15 @@ exit 0
 
 %files
 %{_unitdir}/wsdd.service
-%{_usr}/lib/firewalld/services/wsdd.xml
-%{_usr}/lib/firewalld/services/wsdd-http.xml
 %config(noreplace) %{_sysconfdir}/sysconfig/wsdd
 %{_bindir}/wsdd
 %{_mandir}/man8/wsdd.8*
 %license LICENSE
 %doc AUTHORS README.md
 
+%files firewalld
+%{_usr}/lib/firewalld/services/wsdd.xml
+%{_usr}/lib/firewalld/services/wsdd-http.xml
 
 %changelog
 * Sun Jan 19 2025 Fedora Release Engineering <releng@fedoraproject.org> - 0.8-3
